// bumbag/es/Box.styles.js
import 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import { a as altitude, t as theme } from './getCSSFromStyleObject.js';

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n          align-items: ", ";\n        "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n          justify-content: ", ";\n        "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n          flex-direction: column;\n        "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      display: flex;\n\n      ", "\n\n      ", "\n\n      ", "\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n    ", "\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  margin: unset;\n  padding: unset;\n  border: unset;\n  background: unset;\n  font: unset;\n  font-family: inherit;\n  font-size: 100%;\n  box-sizing: border-box;\n\n  &:focus:not(:focus-visible) {\n    outline: none;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      & {\n        ", " !important;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var FLEX_HORIZONTAL_ALIGN_MAP = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end'
};
var FLEX_VERTICAL_ALIGN_MAP = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end'
};
var style = function style(styleProps) {
  return css(_templateObject(), styleProps.style, getAlignmentAttributes(styleProps), styleProps.altitude && css$1(_templateObject2(), altitude(styleProps.altitude)(styleProps)));
};
var Box = function Box(styleProps) {
  return css(_templateObject3(), theme(styleProps.themeKey, 'styles.base')(styleProps));
};
function getAlignmentAttributes(styleProps) {
  return css$1(_templateObject4(), (styleProps.alignY || styleProps.alignX) && css$1(_templateObject5(), !styleProps.display && css$1(_templateObject6()), styleProps.alignY && css$1(_templateObject7(), FLEX_VERTICAL_ALIGN_MAP[styleProps.alignY]), styleProps.alignX && css$1(_templateObject8(), FLEX_HORIZONTAL_ALIGN_MAP[styleProps.alignX])));
}

var Box_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  style: style,
  Box: Box,
  getAlignmentAttributes: getAlignmentAttributes
});

export { Box_styles as B, Box as a, getAlignmentAttributes as g, style as s };
