// bumbag/es/Card.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { p as palette, c as borderRadius, s as space, t as theme, h as fontWeight, e as fontSize } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-top: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  font-weight: ", ";\n  font-size: ", "em;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-bottom: ", "rem;\n  align-items: center;\n  justify-content: space-between;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  border-radius: ", ";\n  padding: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Card = function Card(styleProps) {
  return css(_templateObject(), palette('background', {
    dark: 'gray900'
  })(styleProps), borderRadius('default')(styleProps), space(3, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardContent = function CardContent(styleProps) {
  return css(_templateObject2(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardHeader = function CardHeader(styleProps) {
  return css(_templateObject3(), space(2, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardTitle = function CardTitle(styleProps) {
  return css(_templateObject4(), fontWeight('semibold')(styleProps), fontSize('300')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardFooter = function CardFooter(styleProps) {
  return css(_templateObject5(), space(3, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Card_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Card: Card,
  CardContent: CardContent,
  CardHeader: CardHeader,
  CardTitle: CardTitle,
  CardFooter: CardFooter
});

export { Card_styles as C, CardContent as a, CardTitle as b, CardHeader as c, CardFooter as d, Card as e };
