// bumbag/es/Dialog.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { c as borderRadius, p as palette, a as altitude, t as theme, s as space, e as fontSize, h as fontWeight } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import { getHiddenScrollbarStyles } from './utils/getHiddenScrollbarStyles.js';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n  max-width: 600px;\n  width: 100%;\n\n  ", ";\n\n  & .bb-DialogContent {\n    max-height: 70vh;\n    overflow-y: ", ";\n\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-right: ", "rem;\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n  & .bb-CalloutContent .bb-Icon {\n    font-size: 1.25em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: ", "rem;\n  right: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      justify-content: flex-end;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      justify-content: space-between;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  align-items: center;\n  background-color: ", ";\n  padding: ", "rem ", "rem;\n\n  ", ";\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  font-size: ", "em;\n  font-weight: ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-bottom: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  border-radius: ", ";\n  background-color: ", ";\n  overflow: hidden;\n  position: relative;\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Dialog = function Dialog(styleProps) {
  return css(_templateObject(), borderRadius('default')(styleProps), palette('background')(styleProps), altitude('100')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogContent = function DialogContent(styleProps) {
  return css(_templateObject2(), space(3, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogHeader = function DialogHeader(styleProps) {
  return css(_templateObject3(), space(2, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogTitle = function DialogTitle(styleProps) {
  return css(_templateObject4(), fontSize('300')(styleProps), fontWeight('semibold')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogFooter = function DialogFooter(styleProps) {
  return css(_templateObject5(), palette('white600', {
    dark: 'black200'
  })(styleProps), space(5, 'minor')(styleProps), space(6, 'minor')(styleProps), styleProps.showActionButtons && styleProps.footer && css$1(_templateObject6()), styleProps.showActionButtons && !styleProps.footer && css$1(_templateObject7()), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogClose = function DialogClose(styleProps) {
  return css(_templateObject8(), space(2, 'major')(styleProps), space(2, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogIconWrapper = function DialogIconWrapper(styleProps) {
  return css(_templateObject9(), space(2, 'major')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogModal = function DialogModal(styleProps) {
  return css(_templateObject10(), altitude('400')(styleProps), styleProps.hasScroll ? 'scroll' : 'visible', getHiddenScrollbarStyles(), theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Dialog_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Dialog: Dialog,
  DialogContent: DialogContent,
  DialogHeader: DialogHeader,
  DialogTitle: DialogTitle,
  DialogFooter: DialogFooter,
  DialogClose: DialogClose,
  DialogIconWrapper: DialogIconWrapper,
  DialogModal: DialogModal
});

export { Dialog_styles as D, Dialog as a, DialogClose as b, DialogContent as c, DialogHeader as d, DialogTitle as e, DialogFooter as f, DialogIconWrapper as g, DialogModal as h };
