// bumbag/es/Link.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { p as palette, h as fontWeight, t as theme } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import { getCapsizeStyles } from './utils/getCapsizeStyles.js';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  ", ";\n\n  display: block;\n\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  fill: ", ";\n  cursor: pointer;\n  font-weight: ", ";\n  text-decoration: none;\n\n  &:hover {\n    text-decoration: underline;\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    & {\n      ", ";\n    }\n  }\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Link = function Link(styleProps) {
  return css(_templateObject(), palette('primary', {
    dark: 'primary300'
  })(styleProps), palette('primary', {
    dark: 'primary300'
  })(styleProps), fontWeight('semibold')(styleProps), theme(styleProps.themeKey, "styles.hover")(styleProps), theme(styleProps.themeKey, "styles.focus")(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var LinkBlock = function LinkBlock(styleProps) {
  return css(_templateObject2(), Link(styleProps), getCapsizeStyles({
    lineHeight: 'default'
  })(styleProps));
};
var LinkInline = function LinkInline(styleProps) {
  return css(_templateObject3(), Link(styleProps));
};

var Link_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Link: Link,
  LinkBlock: LinkBlock,
  LinkInline: LinkInline
});

export { Link_styles as L, Link as a, LinkBlock as b, LinkInline as c };
