// bumbag/es/List.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { s as space, t as theme } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  &&& li {\n    display: inline-block;\n    margin-bottom: unset;\n\n    &:not(:last-child) {\n      margin-right: 1rem;\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  list-style-type: decimal;\n\n  & & {\n    list-style-type: lower-alpha;\n  }\n\n  & & & {\n    list-style-type: lower-roman;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  list-style: unset;\n  list-style-type: none;\n\n  & & {\n    margin-left: ", "rem;\n  }\n\n  & li {\n    &:not(:last-child) {\n      margin-bottom: ", "rem;\n    }\n  }\n\n  & {\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var List = function List(styleProps) {
  return css(_templateObject(), space(4)(styleProps), space(1)(styleProps), styleProps.isOrdered && getOrderedProperties(styleProps), styleProps.orientation === 'horizontal' && getHorizontalProperties(styleProps), theme(styleProps.themeKey, 'styles.base')(styleProps));
};
var ListItem = function ListItem(styleProps) {
  return css(_templateObject2(), theme(styleProps.themeKey, 'styles.base')(styleProps));
};
var getOrderedProperties = function getOrderedProperties(styleProps) {
  return css(_templateObject3(), theme(styleProps.themeKey, 'styles.ordered')(styleProps));
};
var getHorizontalProperties = function getHorizontalProperties(styleProps) {
  return css(_templateObject4(), theme(styleProps.themeKey, 'styles.horizontal')(styleProps));
};

var List_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  List: List,
  ListItem: ListItem,
  getOrderedProperties: getOrderedProperties,
  getHorizontalProperties: getHorizontalProperties
});

export { List_styles as L, ListItem as a, List as b, getHorizontalProperties as c, getOrderedProperties as g };
