// bumbag/es/Pagination.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { t as theme, h as fontWeight } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  font-weight: ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Pagination = function Pagination(styleProps) {
  return css(_templateObject(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PaginationButton = function PaginationButton(styleProps) {
  return css(_templateObject2(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PaginationSelect = function PaginationSelect(styleProps) {
  return css(_templateObject3(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PaginationPrepositionText = function PaginationPrepositionText(styleProps) {
  return css(_templateObject4(), fontWeight('semibold')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Pagination_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Pagination: Pagination,
  PaginationButton: PaginationButton,
  PaginationSelect: PaginationSelect,
  PaginationPrepositionText: PaginationPrepositionText
});

export { Pagination_styles as P, Pagination as a, PaginationButton as b, PaginationSelect as c, PaginationPrepositionText as d };
