// bumbag/es/Radio.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { t as theme, p as palette, h as fontWeight } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';
import { getHiddenInputStyles } from './utils/getHiddenInputStyles.js';

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      border-color: ", ";\n      background: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      border-radius: 50%;\n      content: '';\n      height: 0.5em;\n      left: 50%;\n      position: absolute;\n      top: 50%;\n      transform: translate(-50%, -50%);\n      width: 0.5em;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  && {\n    font-weight: ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  -webkit-appearance: none;\n  background-color: ", ";\n  border: 1px solid ", ";\n  border-radius: 100%;\n  height: 1em;\n  position: relative;\n  min-width: 1em;\n  width: 1em;\n  transition: box-shadow 0.1s ease-in-out 0s, border-color 0.1s, background-color 0.1s;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  && {\n    display: flex;\n    align-items: center;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Radio = function Radio(styleProps) {
  return css(_templateObject(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var RadioIcon = function RadioIcon(styleProps) {
  return css(_templateObject2(), palette('default')(styleProps), palette('white900', {
    dark: 'gray700'
  })(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var RadioLabel = function RadioLabel(styleProps) {
  return css(_templateObject3(), fontWeight('normal')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var HiddenRadio = function HiddenRadio(styleProps) {
  return getHiddenInputStyles({
    iconClassName: 'bb-RadioIcon',
    checkedIconCss: css$1(_templateObject4(), palette('primary', {
      dark: 'primary300'
    })(styleProps)),
    disabledCheckedIconCss: css$1(_templateObject5(), palette('gray300', {
      dark: 'gray'
    })(styleProps), palette('gray300', {
      dark: 'gray'
    })(styleProps)),
    styleProps: styleProps,
    themeKey: 'Radio.Icon'
  });
};
var RadioField = function RadioField(styleProps) {
  return css(_templateObject6(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var RadioGroup = function RadioGroup(styleProps) {
  return css(_templateObject7(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var RadioGroupField = function RadioGroupField(styleProps) {
  return css(_templateObject8(), theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Radio_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Radio: Radio,
  RadioIcon: RadioIcon,
  RadioLabel: RadioLabel,
  HiddenRadio: HiddenRadio,
  RadioField: RadioField,
  RadioGroup: RadioGroup,
  RadioGroupField: RadioGroupField
});

export { HiddenRadio as H, Radio_styles as R, Radio as a, RadioIcon as b, RadioLabel as c, RadioGroup as d, RadioGroupField as e, RadioField as f };
