// bumbag/es/Table.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { p as palette, j as breakpoint, s as space, h as fontWeight, e as fontSize, t as theme, c as borderRadius, a as altitude } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  font-weight: ", ";\n  font-size: ", "rem;\n  text-transform: uppercase;\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      border: 1px solid ", ";\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n  border-radius: ", ";\n  overflow: hidden;\n\n  ", "\n\n  ", "\n\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: ", ";\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n  border-top: 1px solid ", ";\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n  vertical-align: middle;\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  border-bottom: 1px solid ", ";\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n          & thead {\n            display: none;\n          }\n\n          & tbody tr:not(:last-child) {\n            border-bottom: 1px solid ", ";\n          }\n\n          & tbody td {\n            display: block;\n            text-align: left !important;\n            padding: ", "rem ", "rem;\n\n            &::before {\n              display: block;\n              content: attr(data-content);\n              ", ";\n\n              & {\n                ", ";\n              }\n            }\n          }\n\n          & tfoot td {\n            display: block;\n            text-align: left !important;\n          }\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      & tbody tr:nth-child(even) {\n        background-color: ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      & tbody tr:hover {\n        background-color: ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      & tbody tr:not(:last-child) {\n        border-bottom: 1px solid ", ";\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  border-collapse: collapse;\n  border-spacing: 0;\n  text-align: left;\n  width: 100%;\n  color: ", ";\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Table = function Table(styleProps) {
  return css(_templateObject(), palette('text200')(styleProps), styleProps.hasDividers && css$1(_templateObject2(), palette('white800', {
    dark: 'gray900'
  })(styleProps)), styleProps.isHoverable && css$1(_templateObject3(), palette('white600', {
    dark: 'black200'
  })(styleProps)), styleProps.isStriped && css$1(_templateObject4(), palette('white600', {
    dark: 'black200'
  })(styleProps)), styleProps.isResponsive && css$1(_templateObject5(), breakpoint("max-" + styleProps.responsiveBreakpoint, css$1(_templateObject6(), palette('white800', {
    dark: 'gray900'
  })(styleProps), space(2)(styleProps), space(4)(styleProps), tableHeadCellText(styleProps), theme(styleProps.themeKey, "styles.responsive.headCellText")(styleProps), theme(styleProps.themeKey, "styles.responsive.base")(styleProps)))(styleProps)), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableHead = function TableHead(styleProps) {
  return css(_templateObject7(), palette('white800', {
    dark: 'gray900'
  })(styleProps), styleProps.tableVariant !== 'minimal' && css$1(_templateObject8(), palette('white600', {
    dark: 'black200'
  })(styleProps)), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableRow = function TableRow(styleProps) {
  return css(_templateObject9(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableHeadCell = function TableHeadCell(styleProps) {
  return css(_templateObject10(), space(3)(styleProps), space(4)(styleProps), tableHeadCellText(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableCell = function TableCell(styleProps) {
  return css(_templateObject11(), space(3)(styleProps), space(4)(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableFoot = function TableFoot(styleProps) {
  return css(_templateObject12(), palette('white800', {
    dark: 'gray900'
  })(styleProps), styleProps.tableVariant !== 'minimal' && css$1(_templateObject13(), palette('white600', {
    dark: 'black200'
  })(styleProps)), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableBody = function TableBody(styleProps) {
  return css(_templateObject14(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableWrapper = function TableWrapper(styleProps) {
  return css(_templateObject15(), borderRadius('default')(styleProps), styleProps.variant === 'default' && css$1(_templateObject16(), palette('white800', {
    dark: 'gray900'
  })(styleProps)), styleProps.variant === 'shadowed' && css$1(_templateObject17(), altitude('100')(styleProps)), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var tableHeadCellText = function tableHeadCellText(styleProps) {
  return css$1(_templateObject18(), palette('text100')(styleProps), fontWeight('semibold')(styleProps), fontSize('100')(styleProps));
};

var Table_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Table: Table,
  TableHead: TableHead,
  TableRow: TableRow,
  TableHeadCell: TableHeadCell,
  TableCell: TableCell,
  TableFoot: TableFoot,
  TableBody: TableBody,
  TableWrapper: TableWrapper,
  tableHeadCellText: tableHeadCellText
});

export { Table_styles as T, TableWrapper as a, Table as b, TableHead as c, TableCell as d, TableFoot as e, TableHeadCell as f, TableBody as g, TableRow as h, tableHeadCellText as t };
