// bumbag/es/TopNav.styles.js
import 'react';
import './Provider/ThemeContext.js';
import 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import { a as _taggedTemplateLiteralLoose } from './_rollupPluginBabelHelpers.js';
import './ColorMode/ColorModeContext.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import { t as theme, s as space, h as fontWeight, c as borderRadius, p as palette } from './getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import './utils/OutsideClickHandler.js';

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n        box-shadow: inset 0 -2px 0 0 ", ";\n      "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n        background-color: ", ";\n      "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n        color: ", ";\n      "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n        box-shadow: inset 0 -2px 0 0 ", ";\n      "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      fill: ", ";\n\n      ", " & {\n        ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      border-radius: ", ";\n      height: 44px;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      min-height: 2.75rem;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      &&&&&& {\n        cursor: pointer;\n        padding: 0 0.8rem;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  align-items: center;\n  color: inherit;\n  height: 100%;\n  font-weight: ", ";\n  text-decoration: none;\n\n  &&&& {\n    display: flex;\n  }\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  &:hover {\n    ", "\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    outline: none;\n    color: ", ";\n    fill: ", ";\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n\n  &:not(:last-child) {\n    margin-right: ", "rem;\n  }\n\n  &&&&& li {\n    align-items: center;\n    display: flex;\n  }\n\n  &&&&& > *:not(:last-child) {\n    margin-right: ", "rem;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  justify-content: space-between;\n  min-height: 60px;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var TopNav = function TopNav(styleProps) {
  return css(_templateObject(), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TopNavSection = function TopNavSection(styleProps) {
  return css(_templateObject2(), space(2, 'major')(styleProps), space(1, 'minor')(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TopNavItem = function TopNavItem(styleProps) {
  return css(_templateObject3(), fontWeight('semibold')(styleProps), (styleProps.href || styleProps.onClick || styleProps.navId) && css$1(_templateObject4()), styleProps.variant === 'default' && css$1(_templateObject5()), styleProps.variant === 'pill' && css$1(_templateObject6(), borderRadius('default')(styleProps)), styleProps.isActive && css$1(_templateObject7(), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), styleProps.variant === 'default' && css$1(_templateObject8(), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)), theme(styleProps.themeKey, "styles.active")(styleProps)), styleProps.variant !== 'pill' && css$1(_templateObject9(), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)), styleProps.variant === 'pill' && css$1(_templateObject10(), palette('white700', {
    dark: 'black200'
  })(styleProps)), theme(styleProps.themeKey, "styles.hover")(styleProps), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), styleProps.variant === 'default' && css$1(_templateObject11(), palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)), theme(styleProps.themeKey, "styles.focus")(styleProps), theme(styleProps.themeKey, "styles.base")(styleProps));
};

var TopNav_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  TopNav: TopNav,
  TopNavSection: TopNavSection,
  TopNavItem: TopNavItem
});

export { TopNav_styles as T, TopNav as a, TopNavSection as b, TopNavItem as c };
