// bumbag/lib/Alert.styles-4ef2e36c.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject29() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  from {\n    width: 100%;\n  }\n\n  to {\n    width: 0%;\n  }\n"]);

  _templateObject29 = function _templateObject29() {
    return data;
  };

  return data;
}

function _templateObject28() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  from {\n    height: 100%;\n  }\n\n  to {\n    height: 0%;\n  }\n"]);

  _templateObject28 = function _templateObject28() {
    return data;
  };

  return data;
}

function _templateObject27() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  color: ", ";\n"]);

  _templateObject27 = function _templateObject27() {
    return data;
  };

  return data;
}

function _templateObject26() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        animation: ", " ", "ms linear forwards;\n      "]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        animation: ", " ", "ms linear forwards;\n      "]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      ", " ", ";\n    "]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      opacity: 0.3;\n      left: 0;\n    "]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      width: 100%;\n      height: ", ";\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      height: 100%;\n      width: ", ";\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: 0;\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      bottom: 0;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: absolute;\n  background-color: ", ";\n\n  ", "\n\n  ", "\n\n  ", "\n  ", "\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        padding: 0;\n      "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    ", "\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  line-height: 0.9;\n  margin-right: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-top: ", ";\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-bottom: ", ";\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-left: ", ";\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: flex;\n\n      & > *:first-child {\n        margin-right: ", "rem;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  justify-content: space-between;\n  flex: 1;\n  padding: ", "rem ", "rem;\n\n  ", "\n\n  ", "\n  ", "\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: flex;\n\n      & > *:first-child {\n        margin-right: ", "rem;\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: block;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border: 1px solid ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  border-radius: ", ";\n  position: relative;\n  overflow: hidden;\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Alert = function Alert(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette('background', {
    dark: 'gray900'
  })(styleProps), utils_theme.borderRadius('default')(styleProps), styleProps.variant === 'bordered' && core.css(_templateObject2(), utils_theme.palette('white800', {
    dark: 'gray'
  })(styleProps)), styleProps.variant === 'shadowed' && core.css(_templateObject3(), utils_theme.altitude('100')(styleProps)), styleProps.variant === 'tint' && core.css(_templateObject4(), utils_theme.palette(styleProps.type + "Tint", {
    dark: styleProps.type + "Shade"
  })(styleProps)), styleProps.variant === 'fill' && getFillAttributes(styleProps), (styleProps.accent === 'top' || styleProps.accent === 'bottom') && core.css(_templateObject5()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertContent = function AlertContent(styleProps) {
  return emotion.css(_templateObject6(), styleProps.isInline && core.css(_templateObject7(), utils_theme.space(1)(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertWrapper = function AlertWrapper(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.space(1, 'major')(styleProps), utils_theme.space(2, 'major')(styleProps), styleProps.isInline && core.css(_templateObject9(), utils_theme.space(1)(styleProps)), styleProps.accent === true && core.css(_templateObject10(), styleProps.accentSize), styleProps.accent === 'bottom' && core.css(_templateObject11(), styleProps.accentSize), styleProps.accent === 'top' && core.css(_templateObject12(), styleProps.accentSize), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertTitle = function AlertTitle(styleProps) {
  return emotion.css(_templateObject13(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertDescription = function AlertDescription(styleProps) {
  return emotion.css(_templateObject14(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertIconWrapper = function AlertIconWrapper(styleProps) {
  return emotion.css(_templateObject15(), utils_theme.space(4)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertCloseButton = function AlertCloseButton(styleProps) {
  return emotion.css(_templateObject16(), styleProps.isInline && core.css(_templateObject17()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AlertAccent = function AlertAccent(styleProps) {
  return emotion.css(_templateObject18(), utils_theme.palette(styleProps.type)(styleProps), (styleProps.accent === true || styleProps.accent === 'bottom') && core.css(_templateObject19()), styleProps.accent === 'top' && core.css(_templateObject20()), (styleProps.accent === true || styleProps.accent === 'left') && core.css(_templateObject21(), styleProps.accentSize), (styleProps.accent === 'top' || styleProps.accent === 'bottom') && core.css(_templateObject22(), styleProps.accentSize), styleProps.isBackground && core.css(_templateObject23()), !styleProps.isBackground && styleProps.countdown && core.css(_templateObject24(), styleProps.accent === true && core.css(_templateObject25(), heightCountdown, styleProps.countdown), (styleProps.accent === 'top' || styleProps.accent === 'bottom') && core.css(_templateObject26(), widthCountdown, styleProps.countdown)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var getFillAttributes = function getFillAttributes(styleProps) {
  return core.css(_templateObject27(), utils_theme.palette(styleProps.type)(styleProps), utils_theme.palette(styleProps.type + "Inverted")(styleProps));
};
var heightCountdown = core.keyframes(_templateObject28());
var widthCountdown = core.keyframes(_templateObject29());

var Alert_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Alert: Alert,
  AlertContent: AlertContent,
  AlertWrapper: AlertWrapper,
  AlertTitle: AlertTitle,
  AlertDescription: AlertDescription,
  AlertIconWrapper: AlertIconWrapper,
  AlertCloseButton: AlertCloseButton,
  AlertAccent: AlertAccent,
  getFillAttributes: getFillAttributes,
  heightCountdown: heightCountdown,
  widthCountdown: widthCountdown
});

exports.Alert = Alert;
exports.AlertAccent = AlertAccent;
exports.AlertCloseButton = AlertCloseButton;
exports.AlertContent = AlertContent;
exports.AlertDescription = AlertDescription;
exports.AlertIconWrapper = AlertIconWrapper;
exports.AlertTitle = AlertTitle;
exports.AlertWrapper = AlertWrapper;
exports.Alert_styles = Alert_styles;
exports.getFillAttributes = getFillAttributes;
exports.heightCountdown = heightCountdown;
exports.widthCountdown = widthCountdown;
