// bumbag/lib/Autosuggest.styles-0120a646.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
var utils_getHiddenScrollbarStyles = require('./utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  max-height: calc(", " - 1rem);\n  overflow-y: scroll;\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  padding: ", "rem ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  justify-content: center;\n  padding: ", "rem ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n  width: 100%;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    font-weight: 300;\n\n    &[aria-selected=\"true\"] {\n      background-color: ", ";\n    }\n\n    &[aria-disabled=\"true\"] {\n      color: ", ";\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    max-width: 100%;\n    width: 100%;\n    max-height: ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: relative;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Autosuggest = function Autosuggest(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestPopover = function AutosuggestPopover(styleProps) {
  return emotion.css(_templateObject2(), styleProps.popoverHeight, utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestItem = function AutosuggestItem(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.palette('white600')(styleProps), utils_theme.palette('gray100')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestItemText = function AutosuggestItemText(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestStaticItem = function AutosuggestStaticItem(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.space(1.5)(styleProps), utils_theme.space(4)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestClearButtonWrapper = function AutosuggestClearButtonWrapper(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.space(1)(styleProps), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestClearButton = function AutosuggestClearButton(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.space(1)(styleProps), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestInput = function AutosuggestInput(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var AutosuggestField = function AutosuggestField(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ItemsWrapper = function ItemsWrapper(styleProps) {
  return emotion.css(_templateObject10(), styleProps.popoverHeight, utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Autosuggest_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Autosuggest: Autosuggest,
  AutosuggestPopover: AutosuggestPopover,
  AutosuggestItem: AutosuggestItem,
  AutosuggestItemText: AutosuggestItemText,
  AutosuggestStaticItem: AutosuggestStaticItem,
  AutosuggestClearButtonWrapper: AutosuggestClearButtonWrapper,
  AutosuggestClearButton: AutosuggestClearButton,
  AutosuggestInput: AutosuggestInput,
  AutosuggestField: AutosuggestField,
  ItemsWrapper: ItemsWrapper
});

exports.Autosuggest = Autosuggest;
exports.AutosuggestClearButton = AutosuggestClearButton;
exports.AutosuggestClearButtonWrapper = AutosuggestClearButtonWrapper;
exports.AutosuggestField = AutosuggestField;
exports.AutosuggestInput = AutosuggestInput;
exports.AutosuggestItem = AutosuggestItem;
exports.AutosuggestItemText = AutosuggestItemText;
exports.AutosuggestPopover = AutosuggestPopover;
exports.AutosuggestStaticItem = AutosuggestStaticItem;
exports.Autosuggest_styles = Autosuggest_styles;
exports.ItemsWrapper = ItemsWrapper;
