// bumbag/lib/Button.styles-ece38a73.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');
var theme_palette = require('./theme/palette.js');

function _templateObject23() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    background-color: unset;\n    border: unset;\n    color: ", ";\n    fill: ", ";\n    box-shadow: unset;\n\n    &:hover {\n      background-color: rgba(0, 0, 0, 0.05);\n    }\n    &:hover:active {\n      background-color: rgba(0, 0, 0, 0.1);\n    }\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      &:hover {\n        border-color: ", ";\n        background-color: ", ";\n      }\n\n      &:hover:active {\n        border-color: ", ";\n        background-color: ", ";\n      }\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    background-color: ", ";\n    border: 1px solid ", ";\n    color: ", ";\n    fill: ", ";\n    box-shadow: unset;\n\n    ", ";\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    border: 0;\n    background: unset;\n    color: ", ";\n    fill: ", ";\n    box-shadow: unset;\n\n    &:hover {\n      text-decoration: underline;\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    &:hover:active {\n      background-color: ", ";\n\n      & {\n        ", ";\n      }\n    }\n  "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    &:hover {\n      background-color: ", ";\n\n      & {\n        ", ";\n      }\n    }\n  "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-color: transparent;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &:focus {\n    outline: unset;\n    z-index: 2;\n    box-shadow: ", "\n        0px 0px 0px 1px,\n      ", "\n        0px 0px 0px 3px;\n\n    ", ";\n\n    && {\n      ", ";\n    }\n  }\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    cursor: default;\n    outline: unset;\n\n    &:focus {\n      box-shadow: unset !important;\n      outline: unset !important;\n    }\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    cursor: not-allowed;\n    position: relative;\n\n    &:focus {\n      box-shadow: unset !important;\n      outline: unset !important;\n    }\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        font-size: ", "em;\n        min-height: 3.25rem;\n        padding: 0 ", "rem;\n      }\n      & {\n        ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        min-height: 3rem;\n        padding: 0 ", "rem;\n      }\n      & {\n        ", ";\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        ", ";\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        font-size: ", "em;\n        min-height: 2rem;\n        padding: 0 ", "rem;\n      }\n      & {\n        ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    cursor: not-allowed;\n    opacity: ", ";\n    outline: unset;\n    pointer-events: unset;\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    min-height: unset;\n    padding: ", "em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    font-size: 1.25em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: absolute;\n  align-items: center;\n  justify-content: center;\n\n  & + .bb-Text {\n    opacity: 0;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-left: ", "em;\n      margin-right: -", "em;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-left: -", "em;\n      margin-right: ", "em;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", ";\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border: 1px solid ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  background-color: ", ";\n  border-radius: ", ";\n  box-shadow: 0 1px 3px 0px rgba(0, 0, 0, 0.05);\n  color: ", ";\n  fill: ", ";\n  cursor: pointer;\n  display: inline-flex;\n  font-weight: ", ";\n  min-height: 2.75rem;\n  justify-content: center;\n  padding: 0 ", "rem;\n  text-decoration: none;\n  hyphens: auto;\n  transition: box-shadow 0.1s ease-in-out 0s, border 0.1s, background-color 0.1s;\n\n  ", "\n\n  & {\n    ", ";\n  }\n\n  &[disabled],\n  &[aria-disabled=\"true\"] {\n    ", ";\n  }\n\n  ", "\n  ", ";\n  ", ";\n  ", ";\n\n  ", ";\n  ", ";\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var defaultPalette = theme_palette['default']({});
var Button = function Button(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.borderRadius('default')(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.space(4)(styleProps), styleProps.palette === 'default' && core.css(_templateObject2(), utils_theme.palette('white900', {
    dark: 'gray600'
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps), getDisabledProperties(styleProps), styleProps.size && getSizeProperties(styleProps), styleProps.isLoading && getLoadingProperties(styleProps), styleProps.isStatic && getStaticProperties(styleProps), isInteractive(styleProps) && getInteractiveProperties(styleProps), styleProps.variant === 'outlined' && getOutlinedProperties(styleProps), styleProps.variant === 'link' && getLinkProperties(styleProps), styleProps.variant === 'ghost' && getGhostProperties(styleProps));
};
var ButtonIcon = function ButtonIcon(styleProps) {
  return emotion.css(_templateObject3(), styleProps.isBefore && core.css(_templateObject4(), utils_theme.space(1)(styleProps), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.before")(styleProps)), styleProps.isAfter && core.css(_templateObject5(), utils_theme.space(2)(styleProps), utils_theme.space(1)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.after")(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ButtonSpinnerWrapper = function ButtonSpinnerWrapper(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ButtonSpinner = function ButtonSpinner(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ButtonClose = function ButtonClose(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var isInteractive = function isInteractive(styleProps) {
  return !styleProps.isStatic && !styleProps.isLoading && !styleProps.disabled && styleProps.variant !== 'link';
};
var getDisabledProperties = function getDisabledProperties(styleProps) {
  return core.css(_templateObject9(), styleProps.colorMode === 'dark' ? '0.5' : '0.7', utils_theme.theme(styleProps.themeKey, "styles.disabled")(styleProps));
};
var getSizeProperties = function getSizeProperties(styleProps) {
  var styles = {
    small: core.css(_templateObject10(), utils_theme.fontSize('100')(styleProps), utils_theme.space(3)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.small")(styleProps)),
    default: core.css(_templateObject11(), utils_theme.theme(styleProps.themeKey, "styles.sizes.default")(styleProps)),
    medium: core.css(_templateObject12(), utils_theme.space(5)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.medium")(styleProps)),
    large: core.css(_templateObject13(), utils_theme.fontSize('300')(styleProps), utils_theme.space(6)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.large")(styleProps))
  };
  return styles[styleProps.size];
};
var getLoadingProperties = function getLoadingProperties(styleProps) {
  return core.css(_templateObject14(), utils_theme.theme(styleProps.themeKey, "styles.loading")(styleProps));
};
var getStaticProperties = function getStaticProperties(styleProps) {
  return core.css(_templateObject15(), utils_theme.theme(styleProps.themeKey, "styles.static")(styleProps));
};
var getInteractiveProperties = function getInteractiveProperties(styleProps) {
  return core.css(_templateObject16(), utils_theme.palette(styleProps.palette === 'default' ? 'gray200' : styleProps.palette, {
    dark: styleProps.palette === 'default' ? 'gray700' : styleProps.palette
  })(styleProps), utils_theme.palette(styleProps.palette === 'default' ? 'gray100' : styleProps.palette + "200", {
    dark: styleProps.palette === 'default' ? 'gray600' : styleProps.palette + "200"
  })(styleProps), styleProps.palette === 'default' && core.css(_templateObject17()), utils_theme.theme(styleProps.themeKey, "styles.focus")(styleProps), styleProps.variant !== 'link' && core.css(_templateObject18(), utils_theme.palette((styleProps.palette === 'default' ? 'white' : styleProps.palette) + "600", {
    dark: "" + (styleProps.palette === 'default' ? 'black100' : styleProps.palette + "600")
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.hover")(styleProps)), styleProps.variant !== 'link' && core.css(_templateObject19(), utils_theme.palette(styleProps.palette === 'default' ? 'white800' : styleProps.palette + "700", {
    dark: "" + (styleProps.palette === 'default' ? 'black200' : styleProps.palette + "700")
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.hoveractive")(styleProps)));
};
var getLinkProperties = function getLinkProperties(styleProps) {
  return core.css(_templateObject20(), styleProps.palette === 'default' ? utils_theme.palette('text')(styleProps) : utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), styleProps.palette === 'default' ? utils_theme.palette('text')(styleProps) : utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.link")(styleProps));
};
var getOutlinedProperties = function getOutlinedProperties(styleProps) {
  return core.css(_templateObject21(), utils_theme.palette('default')(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), isInteractive(styleProps) && core.css(_templateObject22(), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.palette(styleProps.palette + "Tint", {
    dark: styleProps.palette + "Shade"
  })(styleProps), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.palette(styleProps.palette + "100", {
    dark: styleProps.palette + "900"
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.outlined")(styleProps));
};
var getGhostProperties = function getGhostProperties(styleProps) {
  return core.css(_templateObject23(), styleProps.palette === 'default' ? utils_theme.palette('defaultInverted')(styleProps) : utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), styleProps.palette === 'default' ? utils_theme.palette('defaultInverted')(styleProps) : utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.ghost")(styleProps));
};

var Button_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Button: Button,
  ButtonIcon: ButtonIcon,
  ButtonSpinnerWrapper: ButtonSpinnerWrapper,
  ButtonSpinner: ButtonSpinner,
  ButtonClose: ButtonClose,
  isInteractive: isInteractive,
  getDisabledProperties: getDisabledProperties,
  getSizeProperties: getSizeProperties,
  getLoadingProperties: getLoadingProperties,
  getStaticProperties: getStaticProperties,
  getInteractiveProperties: getInteractiveProperties,
  getLinkProperties: getLinkProperties,
  getOutlinedProperties: getOutlinedProperties,
  getGhostProperties: getGhostProperties
});

exports.Button = Button;
exports.ButtonClose = ButtonClose;
exports.ButtonIcon = ButtonIcon;
exports.ButtonSpinner = ButtonSpinner;
exports.ButtonSpinnerWrapper = ButtonSpinnerWrapper;
exports.Button_styles = Button_styles;
exports.getDisabledProperties = getDisabledProperties;
exports.getGhostProperties = getGhostProperties;
exports.getInteractiveProperties = getInteractiveProperties;
exports.getLinkProperties = getLinkProperties;
exports.getLoadingProperties = getLoadingProperties;
exports.getOutlinedProperties = getOutlinedProperties;
exports.getSizeProperties = getSizeProperties;
exports.getStaticProperties = getStaticProperties;
exports.isInteractive = isInteractive;
