// bumbag/lib/Card.styles-c2e6de71.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  margin-top: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-weight: ", ";\n  font-size: ", "em;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  margin-bottom: ", "rem;\n  align-items: center;\n  justify-content: space-between;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  border-radius: ", ";\n  padding: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Card = function Card(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette('background', {
    dark: 'gray900'
  })(styleProps), utils_theme.borderRadius('default')(styleProps), utils_theme.space(3, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardContent = function CardContent(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardHeader = function CardHeader(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardTitle = function CardTitle(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.fontWeight('semibold')(styleProps), utils_theme.fontSize('300')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var CardFooter = function CardFooter(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.space(3, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Card_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Card: Card,
  CardContent: CardContent,
  CardHeader: CardHeader,
  CardTitle: CardTitle,
  CardFooter: CardFooter
});

exports.Card = Card;
exports.CardContent = CardContent;
exports.CardFooter = CardFooter;
exports.CardHeader = CardHeader;
exports.CardTitle = CardTitle;
exports.Card_styles = Card_styles;
