// bumbag/lib/Container.styles-9dcb1344.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    @media (max-width: ", "px) {\n      max-width: ", "px;\n    }\n\n    @media (max-width: ", "px) {\n      max-width: ", "px;\n    }\n\n    @media (max-width: ", "px) {\n      max-width: ", "px;\n    }\n  "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        max-width: ", "px;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    margin-left: auto;\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    margin-left: auto;\n    margin-right: auto;\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    margin-right: auto;\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      @media (max-width: ", "px) {\n        padding-left: ", "rem;\n        padding-right: ", "rem;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding-left: ", "rem;\n      padding-right: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      max-width: ", "px;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  width: 100%;\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Container = function Container(styleProps) {
  return emotion.css(_templateObject(), !styleProps.isFluid && core.css(_templateObject2(), utils_theme.theme('breakpoints', 'fullHD')(styleProps)), styleProps.isFluid && core.css(_templateObject3(), utils_theme.theme(styleProps.themeKey, 'fluidMargin')(styleProps), utils_theme.theme(styleProps.themeKey, 'fluidMargin')(styleProps), utils_theme.theme(styleProps.themeKey, 'styles.fluid')(styleProps)), (styleProps.isLayout || styleProps.isFluid) && core.css(_templateObject4(), utils_theme.theme('breakpoints.tablet')(styleProps), utils_theme.theme(styleProps.themeKey, 'tabletMargin')(styleProps), utils_theme.theme(styleProps.themeKey, 'tabletMargin')(styleProps)), styleProps.isLayout && core.css(_templateObject5(), utils_theme.theme(styleProps.themeKey, 'styles.layout')(styleProps)), getResponsiveProperties(styleProps), styleProps.align && !styleProps.isFluid ? alignProperties[styleProps.align] : null, utils_theme.theme(styleProps.themeKey, 'styles.base')(styleProps));
};
var alignProperties = {
  left: core.css(_templateObject6()),
  center: core.css(_templateObject7()),
  right: core.css(_templateObject8())
};
function getResponsiveProperties(styleProps) {
  var breakpoint = styleProps.breakpoint,
      isFluid = styleProps.isFluid;
  if (isFluid) return;

  if (breakpoint) {
    return core.css(_templateObject9(), utils_theme.theme('breakpoints', breakpoint)(styleProps));
  }

  return core.css(_templateObject10(), utils_theme.theme('breakpoints', 'fullHD')(styleProps) + 128, utils_theme.theme('breakpoints', 'widescreen')(styleProps), utils_theme.theme('breakpoints', 'widescreen')(styleProps) + 128, utils_theme.theme('breakpoints', 'desktop')(styleProps), utils_theme.theme('breakpoints', 'desktop')(styleProps) + 128, utils_theme.theme('breakpoints', 'tablet')(styleProps));
}

var Container_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Container: Container,
  alignProperties: alignProperties,
  getResponsiveProperties: getResponsiveProperties
});

exports.Container = Container;
exports.Container_styles = Container_styles;
exports.alignProperties = alignProperties;
exports.getResponsiveProperties = getResponsiveProperties;
