// bumbag/lib/Dialog.styles-1146ce34.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
var utils_getHiddenScrollbarStyles = require('./utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  max-width: 600px;\n  width: 100%;\n\n  ", ";\n\n  & .bb-DialogContent {\n    max-height: 70vh;\n    overflow-y: ", ";\n\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  margin-right: ", "rem;\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n  & .bb-CalloutContent .bb-Icon {\n    font-size: 1.25em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: ", "rem;\n  right: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      justify-content: flex-end;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      justify-content: space-between;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  background-color: ", ";\n  padding: ", "rem ", "rem;\n\n  ", ";\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-size: ", "em;\n  font-weight: ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  margin-bottom: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  border-radius: ", ";\n  background-color: ", ";\n  overflow: hidden;\n  position: relative;\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Dialog = function Dialog(styleProps) {
  return emotion.css(_templateObject(), utils_theme.borderRadius('default')(styleProps), utils_theme.palette('background')(styleProps), utils_theme.altitude('100')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogContent = function DialogContent(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.space(3, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogHeader = function DialogHeader(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogTitle = function DialogTitle(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.fontSize('300')(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogFooter = function DialogFooter(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps), utils_theme.space(5, 'minor')(styleProps), utils_theme.space(6, 'minor')(styleProps), styleProps.showActionButtons && styleProps.footer && core.css(_templateObject6()), styleProps.showActionButtons && !styleProps.footer && core.css(_templateObject7()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogClose = function DialogClose(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.space(2, 'major')(styleProps), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogIconWrapper = function DialogIconWrapper(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DialogModal = function DialogModal(styleProps) {
  return emotion.css(_templateObject10(), utils_theme.altitude('400')(styleProps), styleProps.hasScroll ? 'scroll' : 'visible', utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Dialog_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Dialog: Dialog,
  DialogContent: DialogContent,
  DialogHeader: DialogHeader,
  DialogTitle: DialogTitle,
  DialogFooter: DialogFooter,
  DialogClose: DialogClose,
  DialogIconWrapper: DialogIconWrapper,
  DialogModal: DialogModal
});

exports.Dialog = Dialog;
exports.DialogClose = DialogClose;
exports.DialogContent = DialogContent;
exports.DialogFooter = DialogFooter;
exports.DialogHeader = DialogHeader;
exports.DialogIconWrapper = DialogIconWrapper;
exports.DialogModal = DialogModal;
exports.DialogTitle = DialogTitle;
exports.Dialog_styles = Dialog_styles;
