// bumbag/lib/DropdownMenu.styles-2108c433.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');
var Menu_Menu_styles = require('./Menu.styles-ef47d2ca.js');

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", " !important;\n\n  ", "\n\n  padding: ", "rem 0;\n  position: absolute;\n  min-width: 200px;\n  width: unset;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var DropdownMenu = function DropdownMenu(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DropdownMenuPopover = function DropdownMenuPopover(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.altitude(styleProps.altitude)(styleProps), Menu_Menu_styles.Menu(styleProps), utils_theme.space(2)(styleProps));
};
var DropdownMenuItem = function DropdownMenuItem(styleProps) {
  return emotion.css(_templateObject3(), Menu_Menu_styles.MenuItem(styleProps));
};
var DropdownMenuButton = function DropdownMenuButton(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DropdownMenuDisclosure = function DropdownMenuDisclosure(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DropdownMenuItemIcon = function DropdownMenuItemIcon(styleProps) {
  return emotion.css(_templateObject6(), Menu_Menu_styles.MenuItemIcon(styleProps));
};
var DropdownMenuDivider = function DropdownMenuDivider(styleProps) {
  return emotion.css(_templateObject7(), Menu_Menu_styles.MenuDivider(styleProps));
};
var DropdownMenuGroup = function DropdownMenuGroup(styleProps) {
  return emotion.css(_templateObject8(), Menu_Menu_styles.MenuGroup(styleProps));
};
var DropdownMenuGroupTitle = function DropdownMenuGroupTitle(styleProps) {
  return emotion.css(_templateObject9(), Menu_Menu_styles.MenuGroupTitle(styleProps));
};
var DropdownMenuOptionGroup = function DropdownMenuOptionGroup(styleProps) {
  return emotion.css(_templateObject10(), Menu_Menu_styles.MenuOptionGroup(styleProps));
};
var DropdownMenuOptionItem = function DropdownMenuOptionItem(styleProps) {
  return emotion.css(_templateObject11(), Menu_Menu_styles.MenuOptionItem(styleProps));
};
var DropdownMenuOptionItemIconWrapper = function DropdownMenuOptionItemIconWrapper(styleProps) {
  return emotion.css(_templateObject12(), Menu_Menu_styles.MenuOptionItemIconWrapper(styleProps));
};

var DropdownMenu_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  DropdownMenu: DropdownMenu,
  DropdownMenuPopover: DropdownMenuPopover,
  DropdownMenuItem: DropdownMenuItem,
  DropdownMenuButton: DropdownMenuButton,
  DropdownMenuDisclosure: DropdownMenuDisclosure,
  DropdownMenuItemIcon: DropdownMenuItemIcon,
  DropdownMenuDivider: DropdownMenuDivider,
  DropdownMenuGroup: DropdownMenuGroup,
  DropdownMenuGroupTitle: DropdownMenuGroupTitle,
  DropdownMenuOptionGroup: DropdownMenuOptionGroup,
  DropdownMenuOptionItem: DropdownMenuOptionItem,
  DropdownMenuOptionItemIconWrapper: DropdownMenuOptionItemIconWrapper
});

exports.DropdownMenu = DropdownMenu;
exports.DropdownMenuButton = DropdownMenuButton;
exports.DropdownMenuDisclosure = DropdownMenuDisclosure;
exports.DropdownMenuDivider = DropdownMenuDivider;
exports.DropdownMenuGroup = DropdownMenuGroup;
exports.DropdownMenuGroupTitle = DropdownMenuGroupTitle;
exports.DropdownMenuItem = DropdownMenuItem;
exports.DropdownMenuItemIcon = DropdownMenuItemIcon;
exports.DropdownMenuOptionGroup = DropdownMenuOptionGroup;
exports.DropdownMenuOptionItem = DropdownMenuOptionItem;
exports.DropdownMenuOptionItemIconWrapper = DropdownMenuOptionItemIconWrapper;
exports.DropdownMenuPopover = DropdownMenuPopover;
exports.DropdownMenu_styles = DropdownMenu_styles;
