// bumbag/lib/FieldWrapper.styles-c26b9d24.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
var utils_getCapsizeStyles = require('./utils/getCapsizeStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    position: absolute;\n    padding: ", "rem;\n    margin-top: ", "rem;\n    width: max-content;\n    max-width: 400px;\n    z-index: 999999;\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    font-size: ", "rem;\n    padding: ", "rem;\n    min-height: unset;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: block;\n  font-size: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  color: ", ";\n  margin-left: ", "rem;\n  font-weight: ", ";\n  font-size: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-size: ", "rem;\n  color: ", ";\n  margin-left: ", "rem;\n  line-height: ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: block;\n  font-size: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: block;\n  margin-top: ", "rem;\n\n  && {\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    margin-bottom: 0px;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var FieldWrapper = function FieldWrapper(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var Label = function Label(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var DescriptionText = function DescriptionText(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.space(2)(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    fontSize: '150'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var HintText = function HintText(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.fontSize('150')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var OptionalText = function OptionalText(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.fontSize('150')(styleProps), utils_theme.palette('text100')(styleProps), utils_theme.space(2)(styleProps), utils_theme.lineHeight('none')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var RequiredText = function RequiredText(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.palette('danger')(styleProps), utils_theme.space(1)(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.fontSize('150')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ValidationText = function ValidationText(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.fontSize('150')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TooltipTrigger = function TooltipTrigger(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.fontSize('100')(styleProps), utils_theme.space(1)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TooltipPopover = function TooltipPopover(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.space(1, 'major')(styleProps), utils_theme.space(1, 'major')(styleProps), utils_theme.altitude('300')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var FieldWrapper_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  FieldWrapper: FieldWrapper,
  Label: Label,
  DescriptionText: DescriptionText,
  HintText: HintText,
  OptionalText: OptionalText,
  RequiredText: RequiredText,
  ValidationText: ValidationText,
  TooltipTrigger: TooltipTrigger,
  TooltipPopover: TooltipPopover
});

exports.DescriptionText = DescriptionText;
exports.FieldWrapper = FieldWrapper;
exports.FieldWrapper_styles = FieldWrapper_styles;
exports.HintText = HintText;
exports.Label = Label;
exports.OptionalText = OptionalText;
exports.RequiredText = RequiredText;
exports.TooltipPopover = TooltipPopover;
exports.TooltipTrigger = TooltipTrigger;
exports.ValidationText = ValidationText;
