// bumbag/lib/Heading.styles-50040bcb.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
var utils_getCapsizeStyles = require('./utils/getCapsizeStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-weight: ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          & {\n            ", ";\n          }\n        "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          ", ";\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          ", ";\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          ", ";\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          ", ";\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          ", ";\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-family: ", ";\n  font-weight: ", ";\n\n  ", ";\n\n  h1& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n  h2& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n  h3& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n  h4& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n  h5& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n  h6& {\n    ", ";\n\n    & {\n      ", ";\n    }\n\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Heading = function Heading(styleProps) {
  return emotion.css(_templateObject(), utils_theme.font('heading')(styleProps) || utils_theme.font('default')(styleProps), utils_theme.fontWeight('bold')(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h1',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h1.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject2(), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h1',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true,
    shrink: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h1.styles.shrinked")(styleProps)))(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h2',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h2.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject3(), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h2',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true,
    shrink: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h2.styles.shrinked")(styleProps)))(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h3',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h3.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject4(), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h3',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true,
    shrink: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h3.styles.shrinked")(styleProps)))(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h4',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h4.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject5(), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h4',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true,
    shrink: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h4.styles.shrinked")(styleProps)))(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h5',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h5.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject6(), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h5',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true,
    shrink: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h5.styles.shrinked")(styleProps)))(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    themeKey: 'Heading.h6',
    lineHeight: '100',
    fontFamily: 'heading',
    includeBottomGap: true
  })(styleProps), utils_theme.theme(styleProps.themeKey, "h6.styles.base")(styleProps), styleProps.shrinkBelow && utils_theme.breakpoint("max-" + styleProps.shrinkBelow, core.css(_templateObject7(), utils_theme.theme(styleProps.themeKey, "h6.styles.shrinked")(styleProps)))(styleProps), styleProps.isSubHeading && getSubHeadingProperties(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var getSubHeadingProperties = function getSubHeadingProperties(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.fontWeight('semibold')(styleProps), utils_theme.theme(styleProps.themeKey, "subHeading.styles.base")(styleProps));
};

var Heading_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Heading: Heading,
  getSubHeadingProperties: getSubHeadingProperties
});

exports.Heading = Heading;
exports.Heading_styles = Heading_styles;
exports.getSubHeadingProperties = getSubHeadingProperties;
