// bumbag/lib/Input.styles-8ea80a0d.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        ", ";\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & .bb-InputWrapper {\n    flex: 1;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    font-size: inherit;\n    align-items: center;\n    display: flex;\n    position: absolute;\n    height: 2.75em;\n    margin: 0 0.75em;\n    top: 0;\n    right: 0;\n    z-index: 2;\n\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    fill: ", ";\n    height: 2.75em;\n    margin: 0 0.75em;\n    top: 0;\n\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        padding-left: 2.3em;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        padding-right: 2.3em;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        border-color: ", ";\n        box-shadow: ", " 0px 0px 0px\n          3px !important;\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  -webkit-appearance: none;\n  background: ", ";\n  border: 1px solid ", ";\n  border-radius: ", ";\n  color: ", ";\n  height: 2.75em;\n  width: 100%;\n  padding: 0.4em 0.8em;\n  transition: box-shadow 0.1s ease-in-out 0s, border-color 0.1s, background-color 0.1s;\n\n  &[disabled] {\n    background-color: ", ";\n    box-shadow: unset;\n    color: ", ";\n    cursor: not-allowed;\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    outline: unset;\n    z-index: 2;\n    border-color: ", ";\n    box-shadow: ", " 0px 0px 0px 3px !important;\n\n    & {\n      ", ";\n    }\n  }\n\n  &::placeholder {\n    color: ", ";\n\n    & {\n      ", ";\n    }\n  }\n\n  ", "\n\n  ", ";\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  position: relative;\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var InputWrapper = function InputWrapper(styleProps) {
  return emotion.css(_templateObject(), styleProps.size && wrapperSizeProperties(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var Input = function Input(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.palette('default')(styleProps), utils_theme.palette('white900', {
    dark: 'gray700'
  })(styleProps), utils_theme.borderRadius('default')(styleProps), utils_theme.palette('text')(styleProps), utils_theme.palette('white700', {
    dark: 'black200'
  })(styleProps), utils_theme.palette('text100')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.disabled")(styleProps), utils_theme.palette('primary')(styleProps), utils_theme.palette('primaryTint', {
    dark: 'primaryShade'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.focus")(styleProps), utils_theme.palette('gray300')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placeholder")(styleProps), styleProps.state && core.css(_templateObject3(), utils_theme.palette("" + styleProps.state)(styleProps), utils_theme.palette(styleProps.state + "Tint", {
    dark: styleProps.state + "Shade"
  })(styleProps)), (styleProps.after || styleProps.isLoading) && core.css(_templateObject4()), styleProps.before && core.css(_templateObject5()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var InputIcon = function InputIcon(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.palette('text100')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var InputSpinner = function InputSpinner(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var InputField = function InputField(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var InputFieldGroup = function InputFieldGroup(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
function wrapperSizeProperties(styleProps) {
  var properties = {
    small: core.css(_templateObject10(), utils_theme.fontSize('150')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.small")(styleProps)),
    default: core.css(_templateObject11(), utils_theme.theme(styleProps.themeKey, "styles.sizes.default")(styleProps)),
    medium: core.css(_templateObject12(), utils_theme.fontSize('300')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.medium")(styleProps)),
    large: core.css(_templateObject13(), utils_theme.fontSize('400')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.large")(styleProps))
  };
  return properties[styleProps.size];
}

var Input_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  InputWrapper: InputWrapper,
  Input: Input,
  InputIcon: InputIcon,
  InputSpinner: InputSpinner,
  InputField: InputField,
  InputFieldGroup: InputFieldGroup,
  wrapperSizeProperties: wrapperSizeProperties
});

exports.Input = Input;
exports.InputField = InputField;
exports.InputFieldGroup = InputFieldGroup;
exports.InputIcon = InputIcon;
exports.InputSpinner = InputSpinner;
exports.InputWrapper = InputWrapper;
exports.Input_styles = Input_styles;
exports.wrapperSizeProperties = wrapperSizeProperties;
