// bumbag/lib/Link.styles-44d26db0.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
var utils_getCapsizeStyles = require('./utils/getCapsizeStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", ";\n\n  display: block;\n\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  color: ", ";\n  fill: ", ";\n  cursor: pointer;\n  font-weight: ", ";\n  text-decoration: none;\n\n  &:hover {\n    text-decoration: underline;\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    & {\n      ", ";\n    }\n  }\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Link = function Link(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette('primary', {
    dark: 'primary300'
  })(styleProps), utils_theme.palette('primary', {
    dark: 'primary300'
  })(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.hover")(styleProps), utils_theme.theme(styleProps.themeKey, "styles.focus")(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var LinkBlock = function LinkBlock(styleProps) {
  return emotion.css(_templateObject2(), Link(styleProps), utils_getCapsizeStyles.getCapsizeStyles({
    lineHeight: 'default'
  })(styleProps));
};
var LinkInline = function LinkInline(styleProps) {
  return emotion.css(_templateObject3(), Link(styleProps));
};

var Link_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Link: Link,
  LinkBlock: LinkBlock,
  LinkInline: LinkInline
});

exports.Link = Link;
exports.LinkBlock = LinkBlock;
exports.LinkInline = LinkInline;
exports.Link_styles = Link_styles;
