// bumbag/lib/Menu.styles-ef47d2ca.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  color: ", ";\n  font-size: ", "rem;\n  font-weight: ", ";\n  letter-spacing: ", ";\n  padding: ", "rem ", "rem;\n  padding-top: ", "rem;\n  text-transform: uppercase;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  margin-right: ", "rem;\n  width: 16px;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n\n  &[aria-checked=\"true\"] {\n    color: ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & + & {\n    margin-top: ", "rem;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    margin: ", "rem 0;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-left: ", "rem;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-right: ", "rem;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n\n      &&:hover,\n      &&:focus {\n        background-color: ", ";\n      }\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: unset;\n  cursor: pointer;\n  display: block;\n  font-weight: ", ";\n  padding: ", "rem ", "rem;\n  text-align: left;\n  width: 100%;\n  transition: background-color 0.1s;\n\n  a& {\n    color: unset;\n    fill: unset;\n    text-decoration: unset;\n\n    &:hover {\n      color: unset;\n      fill: unset;\n    }\n  }\n\n  &[disabled] {\n    cursor: not-allowed;\n    opacity: 0.5;\n\n    & {\n      ", ";\n    }\n  }\n\n  &:not(:disabled):focus {\n    outline: unset;\n    background-color: ", ";\n\n    & {\n      ", ";\n    }\n  }\n  &:not(:disabled):hover {\n    background-color: ", ";\n\n    & {\n      ", ";\n    }\n  }\n  &:not(:disabled):focus:active,\n  &:not(:disabled):hover:active {\n    background-color: ", ";\n  }\n\n  ", "\n\n  & .bb-Icon {\n    vertical-align: -0.125em;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  width: 100%;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Menu = function Menu(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuItem = function MenuItem(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.fontWeight('semibold')(styleProps), utils_theme.space(1.5)(styleProps), utils_theme.space(4)(styleProps), utils_theme.theme(styleProps.themeKey, "disabled")(styleProps), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "focus")(styleProps), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "hover")(styleProps), utils_theme.palette('white700', {
    dark: 'black300'
  })(styleProps), styleProps.isActive && core.css(_templateObject3(), utils_theme.palette('white700', {
    dark: 'black300'
  })(styleProps), utils_theme.palette('white800', {
    dark: 'black400'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "active")(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuItemIcon = function MenuItemIcon(styleProps) {
  return emotion.css(_templateObject4(), styleProps.isBefore && core.css(_templateObject5(), utils_theme.space(2)(styleProps)), styleProps.isAfter && core.css(_templateObject6(), utils_theme.space(2)(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuDivider = function MenuDivider(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuGroup = function MenuGroup(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuOptionGroup = function MenuOptionGroup(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuOptionItem = function MenuOptionItem(styleProps) {
  return emotion.css(_templateObject10(), utils_theme.palette('primary')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuOptionItemIconWrapper = function MenuOptionItemIconWrapper(styleProps) {
  return emotion.css(_templateObject11(), utils_theme.space(1, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var MenuGroupTitle = function MenuGroupTitle(styleProps) {
  return emotion.css(_templateObject12(), utils_theme.palette('text100')(styleProps), utils_theme.fontSize('100')(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.letterSpacing('400')(styleProps), utils_theme.space(2)(styleProps), utils_theme.space(4)(styleProps), utils_theme.space(1)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Menu_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Menu: Menu,
  MenuItem: MenuItem,
  MenuItemIcon: MenuItemIcon,
  MenuDivider: MenuDivider,
  MenuGroup: MenuGroup,
  MenuOptionGroup: MenuOptionGroup,
  MenuOptionItem: MenuOptionItem,
  MenuOptionItemIconWrapper: MenuOptionItemIconWrapper,
  MenuGroupTitle: MenuGroupTitle
});

exports.Menu = Menu;
exports.MenuDivider = MenuDivider;
exports.MenuGroup = MenuGroup;
exports.MenuGroupTitle = MenuGroupTitle;
exports.MenuItem = MenuItem;
exports.MenuItemIcon = MenuItemIcon;
exports.MenuOptionGroup = MenuOptionGroup;
exports.MenuOptionItem = MenuOptionItem;
exports.MenuOptionItemIconWrapper = MenuOptionItemIconWrapper;
exports.Menu_styles = Menu_styles;
