// bumbag/lib/Modal.styles-6ced861b.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
var utils_getHiddenScrollbarStyles = require('./utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        opacity: 1;\n      "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      opacity: 0;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    transform-origin: ", ";\n    align-items: center;\n    transition-property: ", ";\n    transition-duration: ", ";\n    transition-timing-function: ", ";\n    transition-delay: ", ";\n    transform: ", " ", " !important;\n\n    ", ";\n\n    &[data-enter] {\n      transform: ", " ", " !important;\n\n      ", ";\n    }\n  "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    ", ";\n  "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      bottom: ", "em;\n      left: ", "em;\n      transform: translate3d(0, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      bottom: ", "em;\n      right: ", "em;\n      transform: translate3d(0, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: ", "em;\n      right: ", "em;\n      transform: translate3d(0, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: ", "em;\n      left: ", "em;\n      transform: translate3d(0, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: 50%;\n      right: ", "em;\n      transform: translate3d(0px, -50%, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: 50%;\n      left: ", "em;\n      transform: translate3d(0px, -50%, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      bottom: ", "em;\n      left: 50%;\n      transform: translate3d(-50%, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      top: ", "em;\n      left: 50%;\n      transform: translate3d(-50%, 0, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      left: 50%;\n      top: 50%;\n      transform: translate3d(-50%, -50%, 0px);\n\n      ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: rgba(0, 0, 0, 0.5);\n  position: fixed;\n  top: 0;\n  right: 0;\n  bottom: 0;\n  left: 0;\n  z-index: 19900409;\n\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: fixed;\n  z-index: 19900410;\n  min-width: 320px;\n\n  @media screen and (max-width: 320px) {\n    min-width: unset;\n    width: 100%;\n  }\n\n  ", "\n\n  &:focus {\n    outline: none;\n  }\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Modal = function Modal(styleProps) {
  return emotion.css(_templateObject(), getPlacementAttributes(styleProps), utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ModalContainer = function ModalContainer(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ModalDisclosure = function ModalDisclosure(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var ModalBackdrop = function ModalBackdrop(styleProps) {
  return emotion.css(_templateObject4(), getAnimatedAttributes({})(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var getPlacementAttributes = function getPlacementAttributes(styleProps) {
  var placementAttributes = {
    // @ts-ignore
    center: core.css(_templateObject5(), getAnimatedAttributes({
      transformX: '-50%',
      transformY: '-50%',
      defaultSlide: 'top'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.center")(styleProps)),
    // @ts-ignore
    top: core.css(_templateObject6(), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '-50%',
      transformY: '0px',
      defaultSlide: 'top'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.top")(styleProps)),
    // @ts-ignore
    bottom: core.css(_templateObject7(), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '-50%',
      transformY: '0px',
      defaultSlide: 'bottom'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.bottom")(styleProps)),
    // @ts-ignore
    left: core.css(_templateObject8(), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '-50%',
      defaultSlide: 'left'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.left")(styleProps)),
    // @ts-ignore
    right: core.css(_templateObject9(), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '-50%',
      defaultSlide: 'right'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.right")(styleProps)),
    // @ts-ignore
    'top-start': core.css(_templateObject10(), utils_theme.space(5)(styleProps), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '0px',
      defaultSlide: 'left'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.topStart")(styleProps)),
    // @ts-ignore
    'top-end': core.css(_templateObject11(), utils_theme.space(5)(styleProps), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '0px',
      defaultSlide: 'right'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.topEnd")(styleProps)),
    // @ts-ignore
    'bottom-end': core.css(_templateObject12(), utils_theme.space(5)(styleProps), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '0px',
      defaultSlide: 'bottom'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.bottomEnd")(styleProps)),
    // @ts-ignore
    'bottom-start': core.css(_templateObject13(), utils_theme.space(5)(styleProps), utils_theme.space(5)(styleProps), getAnimatedAttributes({
      transformX: '0px',
      transformY: '0px',
      defaultSlide: 'bottom'
    })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.placements.bottomStart")(styleProps))
  };
  return core.css(_templateObject14(), placementAttributes[styleProps.placement || 'center']);
};

var getSlideOffset = function getSlideOffset(_ref) {
  var position = _ref.position,
      axis = _ref.axis,
      _ref$defaultSlide = _ref.defaultSlide,
      defaultSlide = _ref$defaultSlide === void 0 ? undefined : _ref$defaultSlide,
      _ref$slideOffset = _ref.slideOffset,
      slideOffset = _ref$slideOffset === void 0 ? '100%' : _ref$slideOffset;
  return function (styleProps) {
    var newSlide = typeof styleProps.slide === 'string' ? styleProps.slide : defaultSlide;
    var offset = '';

    if (axis === 'vertical') {
      if (newSlide === 'top') {
        offset = " - " + slideOffset;
      }

      if (newSlide === 'bottom') {
        offset = " + " + slideOffset;
      }
    }

    if (axis === 'horizontal') {
      if (newSlide === 'left') {
        offset = " - " + slideOffset;
      }

      if (newSlide === 'right') {
        offset = " + " + slideOffset;
      }
    }

    return "calc(" + position + offset + ")";
  };
};

var getAnimatedAttributes = function getAnimatedAttributes(opts) {
  return function (styleProps) {
    if (!styleProps.slide && !styleProps.expand && !styleProps.fade) return;
    var transitionPropertyValue = [(styleProps.slide || styleProps.expand) && 'transform', styleProps.fade && 'opacity'].filter(Boolean).join(', ');
    var hiddenTransformX = getSlideOffset({
      position: opts.transformX,
      axis: 'horizontal',
      defaultSlide: opts.defaultSlide,
      slideOffset: opts.slideOffset
    })(styleProps);
    var hiddenTransformY = getSlideOffset({
      position: opts.transformY,
      axis: 'vertical',
      defaultSlide: opts.defaultSlide,
      slideOffset: opts.slideOffset
    })(styleProps);
    var hiddenSlideTransformValue = "translate3d(" + (styleProps.slide ? hiddenTransformX : opts.transformX) + ", " + (styleProps.slide ? hiddenTransformY : opts.transformY) + ", 0px)";
    var hiddenExpandTransformValue = styleProps.expand ? "scale(0.01)" : undefined;
    var hiddenTransformValue = [hiddenSlideTransformValue, hiddenExpandTransformValue].filter(Boolean).join(' ');
    var showTransformValue = "translate3d(" + opts.transformX + ", " + opts.transformY + ", 0px) scale(1)";
    var expandTransformOrigins = {
      center: '50% 50%',
      left: '0% 50%',
      right: '100% 50%',
      top: '50% 0%',
      bottom: '50% 100%'
    };
    return core.css(_templateObject15(), expandTransformOrigins[opts.defaultExpand || styleProps.expand || 'center'], transitionPropertyValue, styleProps.duration || '250ms', styleProps.timingFunction || 'ease-in-out', styleProps.delay || '0s', opts.prevTransformValue, hiddenTransformValue, styleProps.fade && core.css(_templateObject16()), opts.prevTransformValue, showTransformValue, styleProps.fade && core.css(_templateObject17()));
  };
};

var Modal_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Modal: Modal,
  ModalContainer: ModalContainer,
  ModalDisclosure: ModalDisclosure,
  ModalBackdrop: ModalBackdrop,
  getPlacementAttributes: getPlacementAttributes,
  getAnimatedAttributes: getAnimatedAttributes
});

exports.Modal = Modal;
exports.ModalBackdrop = ModalBackdrop;
exports.ModalContainer = ModalContainer;
exports.ModalDisclosure = ModalDisclosure;
exports.Modal_styles = Modal_styles;
exports.getAnimatedAttributes = getAnimatedAttributes;
exports.getPlacementAttributes = getPlacementAttributes;
