// bumbag/lib/Page.styles-99603079.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
var utils_getHiddenScrollbarStyles = require('./utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject30() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject30 = function _templateObject30() {
    return data;
  };

  return data;
}

function _templateObject29() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding-top: ", ";\n    "]);

  _templateObject29 = function _templateObject29() {
    return data;
  };

  return data;
}

function _templateObject28() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject28 = function _templateObject28() {
    return data;
  };

  return data;
}

function _templateObject27() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      position: fixed;\n      width: 100%;\n    "]);

  _templateObject27 = function _templateObject27() {
    return data;
  };

  return data;
}

function _templateObject26() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  min-height: ", ";\n  height: ", ";\n  border-bottom: 1px solid ", ";\n  z-index: 999;\n\n  & > * {\n    height: 100%;\n  }\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  min-height: 100vh;\n  position: relative;\n\n  & .bb-PageWithSidebarSidebarExpandedWrapper {\n    top: ", ";\n  }\n\n  & .bb-PageWithSidebarSidebar {\n    height: calc(100vh - ", ");\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: none;\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose([""]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    min-width: ", ";\n    overflow-y: scroll;\n\n    ", ";\n  }\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: none;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      right: 0;\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: fixed;\n  z-index: 999999;\n  overflow-y: scroll;\n\n  ", ";\n\n  ", "\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      overflow: visible;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-left: 1px solid ", ";\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-right: 1px solid ", ";\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  height: 100vh;\n  min-width: ", ";\n  width: ", ";\n  transform: translateX(0px);\n\n  ", "\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        padding-right: 0px;\n      "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          padding-right: 0px;\n        "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding-right: ", ";\n\n      ", "\n\n      ", "\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        padding-left: 0px;\n      "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          padding-left: 0px;\n        "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding-left: ", ";\n\n      ", "\n\n      ", "\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  width: 100%;\n\n  ", "\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  min-height: 100vh;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding-top: ", "rem;\n      padding-bottom: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      padding: ", "rem ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var PageContent = function PageContent(styleProps) {
  return emotion.css(_templateObject(), utils_theme.space(4, 'major')(styleProps), utils_theme.space(2, 'major')(styleProps), styleProps.isFluid && core.css(_templateObject2(), utils_theme.space(4, 'major')(styleProps), utils_theme.theme('Container.fluidMargin')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.fluid")(styleProps)), utils_theme.breakpoint('max-tablet', core.css(_templateObject3(), utils_theme.space(2, 'major')(styleProps), utils_theme.space(2, 'major')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.mobile")(styleProps)))(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageContentWrapper = function PageContentWrapper(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebar = function PageWithSidebar(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarContent = function PageWithSidebarContent(styleProps) {
  return emotion.css(_templateObject6(), styleProps.sidebarPlacement === 'left' && core.css(_templateObject7(), getWidth(styleProps), utils_theme.breakpoint("max-" + styleProps.collapseBelow, core.css(_templateObject8()))(styleProps), !styleProps.isSidebarOpen && core.css(_templateObject9())), styleProps.sidebarPlacement === 'right' && core.css(_templateObject10(), getWidth(styleProps), utils_theme.breakpoint("max-" + styleProps.collapseBelow, core.css(_templateObject11()))(styleProps), !styleProps.isSidebarOpen && core.css(_templateObject12())), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarSidebar = function PageWithSidebarSidebar(styleProps) {
  return emotion.css(_templateObject13(), utils_theme.palette('background')(styleProps), getWidth(styleProps), getWidth(styleProps), styleProps.sidebarPlacement === 'left' && core.css(_templateObject14(), utils_theme.palette('white800', {
    dark: 'gray700'
  })(styleProps)), styleProps.sidebarPlacement === 'right' && core.css(_templateObject15(), utils_theme.palette('white800', {
    dark: 'gray700'
  })(styleProps)), styleProps.isSidebarMinimized && core.css(_templateObject16()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarSidebarExpandedWrapper = function PageWithSidebarSidebarExpandedWrapper(styleProps) {
  return emotion.css(_templateObject17(), utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), styleProps.sidebarPlacement === 'right' && core.css(_templateObject18()), utils_theme.breakpoint("max-" + styleProps.collapseBelow, core.css(_templateObject19()))(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarSidebarCollapsedWrapper = function PageWithSidebarSidebarCollapsedWrapper(styleProps) {
  return emotion.css(_templateObject20(), styleProps.collapsedSidebarWidth, utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), utils_theme.breakpoint("max-" + styleProps.collapseBelow, core.css(_templateObject21()), {
    else: core.css(_templateObject22())
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarDisclosure = function PageWithSidebarDisclosure(styleProps) {
  return emotion.css(_templateObject23(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithSidebarMinimize = function PageWithSidebarMinimize(styleProps) {
  return emotion.css(_templateObject24(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithHeader = function PageWithHeader(styleProps) {
  return emotion.css(_templateObject25(), styleProps.headerHeight, styleProps.headerHeight, utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithHeaderHeader = function PageWithHeaderHeader(styleProps) {
  return emotion.css(_templateObject26(), utils_theme.palette('background')(styleProps), styleProps.headerHeight, styleProps.headerHeight, utils_theme.palette('white800', {
    dark: 'gray700'
  })(styleProps), styleProps.sticky && core.css(_templateObject27()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithHeaderContent = function PageWithHeaderContent(styleProps) {
  return emotion.css(_templateObject28(), styleProps.sticky && core.css(_templateObject29(), styleProps.isHeaderOpen ? styleProps.headerHeight : 'unset'), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var PageWithHeaderDisclosure = function PageWithHeaderDisclosure(styleProps) {
  return emotion.css(_templateObject30(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

function getWidth(styleProps) {
  if (styleProps.isSidebarMinimized) {
    return styleProps.minimizedSidebarWidth;
  }

  if (styleProps.isCollapsed) {
    return styleProps.collapsedSidebarWidth;
  }

  return styleProps.sidebarWidth;
}

var Page_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  PageContent: PageContent,
  PageContentWrapper: PageContentWrapper,
  PageWithSidebar: PageWithSidebar,
  PageWithSidebarContent: PageWithSidebarContent,
  PageWithSidebarSidebar: PageWithSidebarSidebar,
  PageWithSidebarSidebarExpandedWrapper: PageWithSidebarSidebarExpandedWrapper,
  PageWithSidebarSidebarCollapsedWrapper: PageWithSidebarSidebarCollapsedWrapper,
  PageWithSidebarDisclosure: PageWithSidebarDisclosure,
  PageWithSidebarMinimize: PageWithSidebarMinimize,
  PageWithHeader: PageWithHeader,
  PageWithHeaderHeader: PageWithHeaderHeader,
  PageWithHeaderContent: PageWithHeaderContent,
  PageWithHeaderDisclosure: PageWithHeaderDisclosure
});

exports.PageContent = PageContent;
exports.PageContentWrapper = PageContentWrapper;
exports.PageWithHeader = PageWithHeader;
exports.PageWithHeaderContent = PageWithHeaderContent;
exports.PageWithHeaderDisclosure = PageWithHeaderDisclosure;
exports.PageWithHeaderHeader = PageWithHeaderHeader;
exports.PageWithSidebar = PageWithSidebar;
exports.PageWithSidebarContent = PageWithSidebarContent;
exports.PageWithSidebarDisclosure = PageWithSidebarDisclosure;
exports.PageWithSidebarMinimize = PageWithSidebarMinimize;
exports.PageWithSidebarSidebar = PageWithSidebarSidebar;
exports.PageWithSidebarSidebarCollapsedWrapper = PageWithSidebarSidebarCollapsedWrapper;
exports.PageWithSidebarSidebarExpandedWrapper = PageWithSidebarSidebarExpandedWrapper;
exports.Page_styles = Page_styles;
