// bumbag/lib/Select.styles-92f39526.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      font-size: ", "rem;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    font-size: inherit;\n    align-items: center;\n    display: flex;\n    position: absolute;\n    height: 2.75em;\n    margin: 0 0.75em;\n    top: 0;\n    right: 0;\n    z-index: 2;\n\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    position: absolute;\n    width: 0.8em;\n    height: 2.75em;\n    right: 0.8em;\n    z-index: 1;\n    color: ", ";\n    fill: ", ";\n    pointer-events: none;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  position: relative;\n  width: fit-content;\n\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        border-color: ", ";\n        box-shadow: ", " 0px 0px 0px\n          3px !important;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  appearance: none;\n  background: linear-gradient(", ", ", ");\n  border: 1px solid ", ";\n  border-radius: ", ";\n  color: ", ";\n  height: 2.75em;\n  padding: 0.4em 2em 0.4em 0.8em;\n  line-height: ", ";\n  transition: box-shadow 0.1s ease-in-out 0s, border-color 0.1s, background-color 0.1s;\n\n  &[disabled],\n  &[aria-disabled=\"true\"] {\n    background: ", ";\n    box-shadow: unset;\n\n    & {\n      ", ";\n    }\n  }\n\n  &[disabled] + .bb-Icon {\n    color: ", ";\n    fill: ", ";\n  }\n\n  &:focus {\n    outline: unset;\n    z-index: 2;\n    border-color: ", ";\n    box-shadow: ", " 0px 0px 0px 3px !important;\n  }\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Select = function Select(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette('default')(styleProps), utils_theme.palette('white600', {
    dark: 'gray900'
  })(styleProps), utils_theme.palette('white900', {
    dark: 'gray700'
  })(styleProps), utils_theme.borderRadius('default')(styleProps), styleProps.isPlaceholderSelected ? utils_theme.palette('gray300')(styleProps) : utils_theme.palette('text')(styleProps), utils_theme.lineHeight('default')(styleProps), utils_theme.palette('white700', {
    dark: 'black200'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.disabled")(styleProps), utils_theme.palette('gray400')(styleProps), utils_theme.palette('gray400')(styleProps), utils_theme.palette('primary')(styleProps), utils_theme.palette('primaryTint', {
    dark: 'primaryShade'
  })(styleProps), styleProps.state && core.css(_templateObject2(), utils_theme.palette("" + styleProps.state)(styleProps), utils_theme.palette(styleProps.state + "Tint", {
    dark: styleProps.state + "Shade"
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectWrapper = function SelectWrapper(styleProps) {
  return emotion.css(_templateObject3(), styleProps.size && wrapperSizeProperties(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectIcon = function SelectIcon(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.palette('text')(styleProps), utils_theme.palette('text')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectSpinner = function SelectSpinner(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectField = function SelectField(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
function wrapperSizeProperties(styleProps) {
  var properties = {
    small: core.css(_templateObject7(), utils_theme.fontSize('150')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.small")(styleProps)),
    default: core.css(_templateObject8(), utils_theme.theme(styleProps.themeKey, "styles.sizes.default")(styleProps)),
    medium: core.css(_templateObject9(), utils_theme.fontSize('300')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.medium")(styleProps)),
    large: core.css(_templateObject10(), utils_theme.fontSize('400')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.sizes.large")(styleProps))
  };
  return properties[styleProps.size];
}

var Select_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Select: Select,
  SelectWrapper: SelectWrapper,
  SelectIcon: SelectIcon,
  SelectSpinner: SelectSpinner,
  SelectField: SelectField,
  wrapperSizeProperties: wrapperSizeProperties
});

exports.Select = Select;
exports.SelectField = SelectField;
exports.SelectIcon = SelectIcon;
exports.SelectSpinner = SelectSpinner;
exports.SelectWrapper = SelectWrapper;
exports.Select_styles = Select_styles;
exports.wrapperSizeProperties = wrapperSizeProperties;
