// bumbag/lib/SelectMenu.styles-832ce213.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
var utils_colors = require('./utils/colors.js');
require('deepmerge');
require('lodash/set');
require('capsize');
var utils_getHiddenScrollbarStyles = require('./utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');
var Select_Select_styles = require('./Select.styles-92f39526.js');

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background: ", ";\n  border-bottom: 1px solid ", ";\n  padding: ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & > input {\n    border-color: transparent;\n    border-bottom-color: ", ";\n    border-bottom-left-radius: 0px;\n    border-bottom-right-radius: 0px;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    padding: ", "rem 0;\n    overflow-y: scroll;\n\n    ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          background-color: ", ";\n        "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    font-weight: 300;\n\n    &[aria-selected=\"true\"] {\n      background-color: ", ";\n      color: ", ";\n\n      &:hover {\n        background-color: ", ";\n      }\n\n      ", "\n    }\n\n    &[aria-disabled=\"true\"] {\n      color: ", ";\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n  width: 100%;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  max-width: calc(100% - 4rem);\n  overflow: hidden;\n  text-overflow: ellipsis;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  justify-content: flex-end;\n  min-width: 4rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        border-color: ", ";\n        box-shadow: ", " 0px 0px 0px 3px !important;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", ";\n\n  cursor: default;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 0.4em 0.8em;\n\n  &[aria-expanded=\"true\"] {\n    border-color: ", ";\n    box-shadow: ", " 0px 0px 0px 3px !important;\n  }\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    max-width: 100%;\n    width: 100%;\n    max-height: ", ";\n    padding: 0;\n    display: flex;\n    flex-direction: column;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  position: relative;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var SelectMenu = function SelectMenu(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuPopover = function SelectMenuPopover(styleProps) {
  return emotion.css(_templateObject2(), styleProps.popoverHeight, utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuButton = function SelectMenuButton(styleProps) {
  return emotion.css(_templateObject3(), Select_Select_styles.Select(styleProps), utils_theme.palette('primary')(styleProps), utils_theme.palette('primaryTint')(styleProps), styleProps.state && core.css(_templateObject4(), utils_theme.palette("" + styleProps.state)(styleProps), utils_theme.palette(styleProps.state + "Tint")(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuButtonIcon = function SelectMenuButtonIcon(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuClearButtonWrapper = function SelectMenuClearButtonWrapper(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuClearButton = function SelectMenuClearButton(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuButtonIconsWrapper = function SelectMenuButtonIconsWrapper(styleProps) {
  return emotion.css(_templateObject8(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuButtonText = function SelectMenuButtonText(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuStaticItem = function SelectMenuStaticItem(styleProps) {
  return emotion.css(_templateObject10(), utils_theme.space(1.5)(styleProps), utils_theme.space(4)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuItem = function SelectMenuItem(styleProps) {
  return emotion.css(_templateObject11(), utils_theme.palette('primaryTint')(styleProps), utils_theme.palette('primary900')(styleProps), utils_colors.darken(0.01, 'primaryTint')(styleProps), styleProps.tabIndex === 0 && core.css(_templateObject12(), utils_colors.darken(0.01, 'primaryTint')(styleProps)), utils_theme.palette('gray100')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuItemText = function SelectMenuItemText(styleProps) {
  return emotion.css(_templateObject13(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuItemsWrapper = function SelectMenuItemsWrapper(styleProps) {
  return emotion.css(_templateObject14(), utils_theme.space(2)(styleProps), utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuSearchInputWrapper = function SelectMenuSearchInputWrapper(styleProps) {
  return emotion.css(_templateObject15(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuSearchInput = function SelectMenuSearchInput(styleProps) {
  return emotion.css(_templateObject16(), utils_theme.palette('white800')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuTagsWrapper = function SelectMenuTagsWrapper(styleProps) {
  return emotion.css(_templateObject17(), utils_theme.palette('white600')(styleProps), utils_theme.palette('white800')(styleProps), utils_theme.space(2)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SelectMenuField = function SelectMenuField(styleProps) {
  return emotion.css(_templateObject18(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var SelectMenu_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  SelectMenu: SelectMenu,
  SelectMenuPopover: SelectMenuPopover,
  SelectMenuButton: SelectMenuButton,
  SelectMenuButtonIcon: SelectMenuButtonIcon,
  SelectMenuClearButtonWrapper: SelectMenuClearButtonWrapper,
  SelectMenuClearButton: SelectMenuClearButton,
  SelectMenuButtonIconsWrapper: SelectMenuButtonIconsWrapper,
  SelectMenuButtonText: SelectMenuButtonText,
  SelectMenuStaticItem: SelectMenuStaticItem,
  SelectMenuItem: SelectMenuItem,
  SelectMenuItemText: SelectMenuItemText,
  SelectMenuItemsWrapper: SelectMenuItemsWrapper,
  SelectMenuSearchInputWrapper: SelectMenuSearchInputWrapper,
  SelectMenuSearchInput: SelectMenuSearchInput,
  SelectMenuTagsWrapper: SelectMenuTagsWrapper,
  SelectMenuField: SelectMenuField
});

exports.SelectMenu = SelectMenu;
exports.SelectMenuButton = SelectMenuButton;
exports.SelectMenuButtonIcon = SelectMenuButtonIcon;
exports.SelectMenuButtonIconsWrapper = SelectMenuButtonIconsWrapper;
exports.SelectMenuButtonText = SelectMenuButtonText;
exports.SelectMenuClearButton = SelectMenuClearButton;
exports.SelectMenuClearButtonWrapper = SelectMenuClearButtonWrapper;
exports.SelectMenuField = SelectMenuField;
exports.SelectMenuItem = SelectMenuItem;
exports.SelectMenuItemText = SelectMenuItemText;
exports.SelectMenuItemsWrapper = SelectMenuItemsWrapper;
exports.SelectMenuPopover = SelectMenuPopover;
exports.SelectMenuSearchInput = SelectMenuSearchInput;
exports.SelectMenuSearchInputWrapper = SelectMenuSearchInputWrapper;
exports.SelectMenuStaticItem = SelectMenuStaticItem;
exports.SelectMenuTagsWrapper = SelectMenuTagsWrapper;
exports.SelectMenu_styles = SelectMenu_styles;
