// bumbag/lib/SideNav.styles-af754712.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n      box-shadow: inset 3px 0 0 0 ", ";\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  color: ", ";\n  cursor: pointer;\n  display: flex;\n  min-height: 2.75em;\n  padding: 0 1rem;\n  padding-left: ", "rem;\n  transition: box-shadow 0.1s ease-in-out 0s, background-color 0.1s, color 0.1s;\n\n  a& {\n    color: ", ";\n    fill: ", ";\n    text-decoration: unset;\n\n    &:hover {\n      color: ", ";\n      fill: ", ";\n    }\n  }\n\n  & {\n    ", ";\n  }\n\n  &&&:hover {\n    color: ", ";\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    outline: unset;\n    background-color: ", ";\n\n    & {\n      ", ";\n    }\n  }\n\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  font-size: ", "rem;\n  font-weight: ", ";\n  margin-bottom: ", "rem;\n  padding-left: ", "rem;\n  text-transform: uppercase;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      margin-bottom: ", "rem;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    margin-left: unset;\n  }\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  width: 100%;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var SideNav = function SideNav(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SideNavLevel = function SideNavLevel(styleProps) {
  return emotion.css(_templateObject2(), styleProps.level === 0 && core.css(_templateObject3(), utils_theme.space(3, 'major')(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SideNavLevelTitle = function SideNavLevelTitle(styleProps) {
  return emotion.css(_templateObject4(), utils_theme.fontSize('150')(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.space(2)(styleProps), styleProps.level + 1, utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SideNavItem = function SideNavItem(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.palette('text200')(styleProps), styleProps.level === 1 ? utils_theme.space(2, 'major')(styleProps) : styleProps.level, utils_theme.palette('text200')(styleProps), utils_theme.palette('text200')(styleProps), utils_theme.palette('text200')(styleProps), utils_theme.palette('text200')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps), utils_theme.palette('primary', {
    dark: 'primary300'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.hover")(styleProps), utils_theme.palette('primaryTint', {
    dark: 'primaryShade'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.focus")(styleProps), styleProps.isActive && core.css(_templateObject6(), utils_theme.palette('primaryTint', {
    dark: 'primaryShade'
  })(styleProps), utils_theme.palette('primary')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.active")(styleProps)));
};

var SideNav_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  SideNav: SideNav,
  SideNavLevel: SideNavLevel,
  SideNavLevelTitle: SideNavLevelTitle,
  SideNavItem: SideNavItem
});

exports.SideNav = SideNav;
exports.SideNavItem = SideNavItem;
exports.SideNavLevel = SideNavLevel;
exports.SideNavLevelTitle = SideNavLevelTitle;
exports.SideNav_styles = SideNav_styles;
