// bumbag/lib/Switch.styles-96991449.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');
var utils_getHiddenInputStyles = require('./utils/getHiddenInputStyles.js');

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background: ", ";\n      content: '';\n      border-radius: 100%;\n      border: 1px solid ", ";\n      height: 1em;\n      width: 1em;\n      top: 0.2em;\n      left: 0.2em;\n      transition: all ease 0.2s;\n      position: absolute;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-color: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-color: ", ";\n      left: 1.25em;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n      border-color: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n      transition: all ease 0.2s;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    font-weight: ", ";\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  background-color: ", ";\n  border: 1px solid ", ";\n  border-radius: 1em;\n  height: 1.5em;\n  position: relative;\n  width: 2.5em;\n  transition: box-shadow 0.1s ease-in-out 0s, border-color 0.1s, background-color 0.1s;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  && {\n    display: flex;\n    align-items: center;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Switch = function Switch(styleProps) {
  return emotion.css(_templateObject(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SwitchIcon = function SwitchIcon(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.palette('default')(styleProps), utils_theme.palette('gray100', {
    dark: 'gray700'
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SwitchLabel = function SwitchLabel(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.fontWeight('normal')(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var HiddenSwitch = function HiddenSwitch(styleProps) {
  return utils_getHiddenInputStyles.getHiddenInputStyles({
    iconClassName: 'bb-SwitchIcon',
    checkedCss: core.css(_templateObject4(), utils_theme.palette(styleProps.palette || 'primary', {
      dark: (styleProps.palette || 'primary') + "700"
    })(styleProps)),
    disabledCheckedCss: core.css(_templateObject5(), utils_theme.palette((styleProps.palette || 'primary') + "100", {
      dark: (styleProps.palette || 'primary') + "800"
    })(styleProps), utils_theme.palette((styleProps.palette || 'primary') + "100", {
      dark: (styleProps.palette || 'primary') + "800"
    })(styleProps)),
    disabledUncheckedIconCss: core.css(_templateObject6(), utils_theme.palette('white700')(styleProps)),
    checkedIconCss: core.css(_templateObject7(), utils_theme.palette(styleProps.palette || 'primary', {
      dark: (styleProps.palette || 'primary') + "700"
    })(styleProps)),
    disabledCheckedIconCss: core.css(_templateObject8(), utils_theme.palette((styleProps.palette || 'primary') + "100", {
      dark: (styleProps.palette || 'primary') + "800"
    })(styleProps)),
    uncheckedIconCss: core.css(_templateObject9(), utils_theme.palette('default', {
      dark: 'gray100'
    })(styleProps), utils_theme.palette('gray100', {
      dark: 'gray700'
    })(styleProps)),
    styleProps: styleProps,
    themeKey: 'Switch.Icon'
  });
};
var SwitchGroup = function SwitchGroup(styleProps) {
  return emotion.css(_templateObject10(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SwitchField = function SwitchField(styleProps) {
  return emotion.css(_templateObject11(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var SwitchGroupField = function SwitchGroupField(styleProps) {
  return emotion.css(_templateObject12(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

var Switch_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Switch: Switch,
  SwitchIcon: SwitchIcon,
  SwitchLabel: SwitchLabel,
  HiddenSwitch: HiddenSwitch,
  SwitchGroup: SwitchGroup,
  SwitchField: SwitchField,
  SwitchGroupField: SwitchGroupField
});

exports.HiddenSwitch = HiddenSwitch;
exports.Switch = Switch;
exports.SwitchField = SwitchField;
exports.SwitchGroup = SwitchGroup;
exports.SwitchGroupField = SwitchGroupField;
exports.SwitchIcon = SwitchIcon;
exports.SwitchLabel = SwitchLabel;
exports.Switch_styles = Switch_styles;
