// bumbag/lib/Table.styles-c23cd6ec.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  color: ", ";\n  font-weight: ", ";\n  font-size: ", "rem;\n  text-transform: uppercase;\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border: 1px solid ", ";\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  border-radius: ", ";\n  overflow: hidden;\n\n  ", "\n\n  ", "\n\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  border-top: 1px solid ", ";\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  padding: ", "rem ", "rem;\n  vertical-align: middle;\n  ", ";\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    ", ";\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      background-color: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  border-bottom: 1px solid ", ";\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n          & thead {\n            display: none;\n          }\n\n          & tbody tr:not(:last-child) {\n            border-bottom: 1px solid ", ";\n          }\n\n          & tbody td {\n            display: block;\n            text-align: left !important;\n            padding: ", "rem ", "rem;\n\n            &::before {\n              display: block;\n              content: attr(data-content);\n              ", ";\n\n              & {\n                ", ";\n              }\n            }\n          }\n\n          & tfoot td {\n            display: block;\n            text-align: left !important;\n          }\n\n          & {\n            ", ";\n          }\n        "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & tbody tr:nth-child(even) {\n        background-color: ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & tbody tr:hover {\n        background-color: ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & tbody tr:not(:last-child) {\n        border-bottom: 1px solid ", ";\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  border-collapse: collapse;\n  border-spacing: 0;\n  text-align: left;\n  width: 100%;\n  color: ", ";\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Table = function Table(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette('text200')(styleProps), styleProps.hasDividers && core.css(_templateObject2(), utils_theme.palette('white800', {
    dark: 'gray900'
  })(styleProps)), styleProps.isHoverable && core.css(_templateObject3(), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps)), styleProps.isStriped && core.css(_templateObject4(), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps)), styleProps.isResponsive && core.css(_templateObject5(), utils_theme.breakpoint("max-" + styleProps.responsiveBreakpoint, core.css(_templateObject6(), utils_theme.palette('white800', {
    dark: 'gray900'
  })(styleProps), utils_theme.space(2)(styleProps), utils_theme.space(4)(styleProps), tableHeadCellText(styleProps), utils_theme.theme(styleProps.themeKey, "styles.responsive.headCellText")(styleProps), utils_theme.theme(styleProps.themeKey, "styles.responsive.base")(styleProps)))(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableHead = function TableHead(styleProps) {
  return emotion.css(_templateObject7(), utils_theme.palette('white800', {
    dark: 'gray900'
  })(styleProps), styleProps.tableVariant !== 'minimal' && core.css(_templateObject8(), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableRow = function TableRow(styleProps) {
  return emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableHeadCell = function TableHeadCell(styleProps) {
  return emotion.css(_templateObject10(), utils_theme.space(3)(styleProps), utils_theme.space(4)(styleProps), tableHeadCellText(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableCell = function TableCell(styleProps) {
  return emotion.css(_templateObject11(), utils_theme.space(3)(styleProps), utils_theme.space(4)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableFoot = function TableFoot(styleProps) {
  return emotion.css(_templateObject12(), utils_theme.palette('white800', {
    dark: 'gray900'
  })(styleProps), styleProps.tableVariant !== 'minimal' && core.css(_templateObject13(), utils_theme.palette('white600', {
    dark: 'black200'
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableBody = function TableBody(styleProps) {
  return emotion.css(_templateObject14(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TableWrapper = function TableWrapper(styleProps) {
  return emotion.css(_templateObject15(), utils_theme.borderRadius('default')(styleProps), styleProps.variant === 'default' && core.css(_templateObject16(), utils_theme.palette('white800', {
    dark: 'gray900'
  })(styleProps)), styleProps.variant === 'shadowed' && core.css(_templateObject17(), utils_theme.altitude('100')(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var tableHeadCellText = function tableHeadCellText(styleProps) {
  return core.css(_templateObject18(), utils_theme.palette('text100')(styleProps), utils_theme.fontWeight('semibold')(styleProps), utils_theme.fontSize('100')(styleProps));
};

var Table_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Table: Table,
  TableHead: TableHead,
  TableRow: TableRow,
  TableHeadCell: TableHeadCell,
  TableCell: TableCell,
  TableFoot: TableFoot,
  TableBody: TableBody,
  TableWrapper: TableWrapper,
  tableHeadCellText: tableHeadCellText
});

exports.Table = Table;
exports.TableBody = TableBody;
exports.TableCell = TableCell;
exports.TableFoot = TableFoot;
exports.TableHead = TableHead;
exports.TableHeadCell = TableHeadCell;
exports.TableRow = TableRow;
exports.TableWrapper = TableWrapper;
exports.Table_styles = Table_styles;
exports.tableHeadCellText = tableHeadCellText;
