// bumbag/lib/Tabs.styles-baf33aed.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject31() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      justify-content: flex-end;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject31 = function _templateObject31() {
    return data;
  };

  return data;
}

function _templateObject30() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      justify-content: center;\n\n      & {\n        ", ";\n      }\n    "]);

  _templateObject30 = function _templateObject30() {
    return data;
  };

  return data;
}

function _templateObject29() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & {\n        justify-content: start;\n\n        ", ";\n      }\n    "]);

  _templateObject29 = function _templateObject29() {
    return data;
  };

  return data;
}

function _templateObject28() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &:focus {\n    outline: none;\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject28 = function _templateObject28() {
    return data;
  };

  return data;
}

function _templateObject27() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n            border-bottom: 1px solid ", ";\n          "]);

  _templateObject27 = function _templateObject27() {
    return data;
  };

  return data;
}

function _templateObject26() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n            border-right: 1px solid ", ";\n          "]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      flex-direction: column;\n    "]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  display: flex;\n  font-weight: ", ";\n\n  ", "\n\n  ", "\n\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: flex;\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      & .bb-TabsTab {\n        flex: 1;\n      }\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  width: 100%;\n\n  ", "\n\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        background-color: ", ";\n      "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        color: ", ";\n      "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        box-shadow: ", " 0px 0px 0px 1px,\n          ", " 0px 0px 0px 3px;\n      "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              box-shadow: inset 0 -2px 0 0 ", ";\n            "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              box-shadow: inset -2px 0 0 0 ", ";\n            "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        ", ";\n      "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        background-color: ", ";\n        color: ", ";\n      "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              border-bottom-color: ", ";\n            "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              border-right-color: ", ";\n            "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        border: 1px solid ", ";\n\n        ", ";\n      "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              box-shadow: inset 0 -2px 0 0 ", ";\n            "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n              box-shadow: inset -2px 0 0 0 ", ";\n            "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        ", ";\n      "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n            border-top-left-radius: ", ";\n            border-top-right-radius: ", ";\n\n            &:not(:last-child) {\n              margin-right: 0.5rem;\n            }\n          "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n            width: 100%;\n            border-top-left-radius: ", ";\n            border-bottom-left-radius: ", ";\n            margin-right: -2px;\n\n            &:not(:last-child) {\n              margin-bottom: 0.5rem;\n            }\n          "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border-radius: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      border: 1px solid transparent;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  cursor: pointer;\n  display: flex;\n  height: 2.75em;\n  justify-content: center;\n  margin-bottom: -1px;\n  padding: 0 1rem;\n  transition: box-shadow 0.1s ease-in-out 0s, background-color 0.1s, color 0.1s;\n\n  ", "\n\n  ", "\n\n  ", "\n\n    ", "\n\n  &[aria-selected='true'] {\n    color: ", ";\n    fill: ", ";\n\n    ", "\n\n    ", "\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n\n  &:focus {\n    outline: unset;\n\n    ", "\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n\n  &:not([aria-selected='true']):hover {\n    ", "\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Tab = function Tab(styleProps) {
  return emotion.css(_templateObject(), styleProps.disabled && core.css(_templateObject2(), utils_theme.palette('gray300')(styleProps)), styleProps.variant === 'boxed' && core.css(_templateObject3()), styleProps.variant === 'button' && core.css(_templateObject4(), utils_theme.borderRadius('default')(styleProps)), styleProps.orientation === 'vertical' ? core.css(_templateObject5(), utils_theme.borderRadius('default')(styleProps), utils_theme.borderRadius('default')(styleProps)) : core.css(_templateObject6(), utils_theme.borderRadius('default')(styleProps), utils_theme.borderRadius('default')(styleProps)), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), styleProps.variant === 'default' && core.css(_templateObject7(), styleProps.orientation === 'vertical' ? core.css(_templateObject8(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)) : core.css(_templateObject9(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps))), styleProps.variant === 'boxed' && core.css(_templateObject10(), utils_theme.palette('white900', {
    dark: 'gray700'
  })(styleProps), styleProps.orientation === 'vertical' ? core.css(_templateObject11(), utils_theme.palette('white', {
    dark: 'background'
  })(styleProps)) : core.css(_templateObject12(), utils_theme.palette('white', {
    dark: 'background'
  })(styleProps))), styleProps.variant === 'button' && core.css(_templateObject13(), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.selected")(styleProps), styleProps.variant === 'default' && core.css(_templateObject14(), styleProps.orientation === 'vertical' ? core.css(_templateObject15(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)) : core.css(_templateObject16(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps))), (styleProps.variant === 'boxed' || styleProps.variant === 'button') && core.css(_templateObject17(), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.palette(styleProps.palette + "200")(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.focus")(styleProps), styleProps.variant !== 'button' && core.css(_templateObject18(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps)), styleProps.variant === 'button' && core.css(_templateObject19(), utils_theme.palette('white700', {
    dark: 'black200'
  })(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.hover")(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var Tabs = function Tabs(styleProps) {
  return emotion.css(_templateObject20(), styleProps.isFitted && core.css(_templateObject21()), styleProps.orientation === 'vertical' && core.css(_templateObject22()), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TabsList = function TabsList(styleProps) {
  return emotion.css(_templateObject23(), utils_theme.fontWeight('semibold')(styleProps), styleProps.orientation === 'vertical' && core.css(_templateObject24()), (styleProps.variant === 'boxed' || styleProps.variant === 'default') && core.css(_templateObject25(), styleProps.orientation === 'vertical' ? core.css(_templateObject26(), utils_theme.palette('white900', {
    dark: 'gray700'
  })(styleProps)) : core.css(_templateObject27(), utils_theme.palette('white900', {
    dark: 'gray700'
  })(styleProps))), getAlignAttributes(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TabsPanel = function TabsPanel(styleProps) {
  return emotion.css(_templateObject28(), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};

function getAlignAttributes(styleProps) {
  var sizeAttributes = {
    left: core.css(_templateObject29(), utils_theme.theme(styleProps.themeKey, "styles.align.left")(styleProps)),
    center: core.css(_templateObject30(), utils_theme.theme(styleProps.themeKey, "styles.align.center")(styleProps)),
    right: core.css(_templateObject31(), utils_theme.theme(styleProps.themeKey, "styles.align.right")(styleProps))
  };
  return sizeAttributes[styleProps.align || 'left'];
}

var Tabs_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Tab: Tab,
  Tabs: Tabs,
  TabsList: TabsList,
  TabsPanel: TabsPanel
});

exports.Tab = Tab;
exports.Tabs = Tabs;
exports.TabsList = TabsList;
exports.TabsPanel = TabsPanel;
exports.Tabs_styles = Tabs_styles;
