// bumbag/lib/Tag.styles-c8ce6208.js
'use strict';

require('react');
require('./Provider/ThemeContext.js');
require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-41d6b37b.js');
require('./ColorMode/ColorModeContext.js');
require('lodash/kebabCase');
var utils_theme = require('./getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
var utils_colors = require('./utils/colors.js');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('./utils/OutsideClickHandler.js');

function _templateObject9() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    font-size: 1.25em;\n    height: 2.5rem;\n    & {\n      ", ";\n    }\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    font-size: 1em;\n    height: 2rem;\n    & {\n      ", ";\n    }\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n    & {\n      ", ";\n    }\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    background-color: ", ";\n    color: ", ";\n    fill: ", ";\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  & {\n    background-color: unset;\n    border: 1px solid ", ";\n    color: ", ";\n    fill: ", ";\n  }\n  & {\n    ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n        background-color: unset;\n        color: ", ";\n\n        &:hover {\n          background-color: ", ";\n        }\n      "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  &&& {\n    background-color: ", ";\n    color: ", ";\n    fill: ", ";\n    height: 1.5em;\n    padding: 0 ", "em;\n    margin-right: ", "em;\n    margin-left: -", "em;\n\n    &:hover {\n      background-color: ", ";\n    }\n\n    &:focus {\n      box-shadow: unset;\n    }\n\n    ", "\n\n    & {\n      ", ";\n    }\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  height: inherit;\n  padding: 0 ", "em;\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  align-items: center;\n  background-color: ", ";\n  border-radius: ", ";\n  color: ", ";\n  display: inline-flex;\n  fill: ", ";\n  font-size: ", "rem;\n  font-weight: ", ";\n  height: 1.5rem;\n  justify-content: center;\n\n  ", "\n  ", "\n  ", "\n\n  & {\n    ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Tag = function Tag(styleProps) {
  return emotion.css(_templateObject(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette === 'text' ? 'gray900' : styleProps.palette
  })(styleProps), utils_theme.borderRadius('default')(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.fontSize('100')(styleProps), utils_theme.fontWeight('semibold')(styleProps), styleProps.variant === 'outlined' && outlinedProperties(styleProps), styleProps.variant === 'tint' && tintProperties(styleProps), styleProps.size && sizeProperties(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TagContent = function TagContent(styleProps) {
  return emotion.css(_templateObject2(), utils_theme.space(3)(styleProps), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var TagClose = function TagClose(styleProps) {
  return emotion.css(_templateObject3(), utils_theme.palette(styleProps.palette)(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.palette(styleProps.palette + "Inverted")(styleProps), utils_theme.space(1)(styleProps), utils_theme.space(1)(styleProps), utils_theme.space(1)(styleProps), utils_colors.darken(0.1, styleProps.palette)(styleProps), styleProps.variant === 'outlined' && core.css(_templateObject4(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette('white700')(styleProps)), utils_theme.theme(styleProps.themeKey, "styles.base")(styleProps));
};
var outlinedProperties = function outlinedProperties(styleProps) {
  return emotion.css(_templateObject5(), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.palette(styleProps.palette, {
    dark: styleProps.palette + "300"
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.outlined")(styleProps));
};
var tintProperties = function tintProperties(styleProps) {
  return emotion.css(_templateObject6(), utils_theme.palette(styleProps.palette + "Tint", {
    dark: styleProps.palette === 'text' ? 'gray900' : styleProps.palette + "Shade"
  })(styleProps), utils_theme.palette(styleProps.palette + "TintInverted", {
    dark: styleProps.palette === 'text' ? 'white' : styleProps.palette + "ShadeInverted"
  })(styleProps), utils_theme.palette(styleProps.palette + "TintInverted", {
    dark: styleProps.palette === 'text' ? 'white' : styleProps.palette + "ShadeInverted"
  })(styleProps), utils_theme.theme(styleProps.themeKey, "styles.outlined")(styleProps));
};
var sizeProperties = function sizeProperties(styleProps) {
  var sizes = {
    default: emotion.css(_templateObject7(), utils_theme.theme(styleProps.themeKey, "styles.sizes.default")(styleProps)),
    medium: emotion.css(_templateObject8(), utils_theme.theme(styleProps.themeKey, "styles.sizes.medium")(styleProps)),
    large: emotion.css(_templateObject9(), utils_theme.theme(styleProps.themeKey, "styles.sizes.large")(styleProps))
  };
  return sizes[styleProps.size];
};

var Tag_styles = /*#__PURE__*/Object.freeze({
  __proto__: null,
  Tag: Tag,
  TagContent: TagContent,
  TagClose: TagClose,
  outlinedProperties: outlinedProperties,
  tintProperties: tintProperties,
  sizeProperties: sizeProperties
});

exports.Tag = Tag;
exports.TagClose = TagClose;
exports.TagContent = TagContent;
exports.Tag_styles = Tag_styles;
exports.outlinedProperties = outlinedProperties;
exports.sizeProperties = sizeProperties;
exports.tintProperties = tintProperties;
