// bumbag/lib/Tabs/Tabs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
require('../utils/OutsideClickHandler.js');
var reakit = require('reakit');
require('../Box.styles-fefb7baf.js');
var Box_Box = require('../Box/Box.js');
var Tabs_Tabs_styles = require('../Tabs.styles-baf33aed.js');
var Tabs_TabsState = require('./TabsState.js');

var TabsContext = React.createContext({
  tabs: {},
  overrides: {}
});
var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var baseId = props.baseId,
      children = props.children,
      loop = props.loop,
      manual = props.manual,
      orientation = props.orientation,
      overrides = props.overrides,
      selectedId = props.selectedId,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["baseId", "children", "loop", "manual", "orientation", "overrides", "selectedId"]);

  var boxProps = Box_Box.Box.useProps(restProps);
  var tabs = Tabs_TabsState.useTabState({
    baseId: baseId,
    loop: loop,
    manual: manual,
    orientation: orientation,
    selectedId: selectedId
  });
  var className = utils_useClassName.useClassName({
    style: Tabs_Tabs_styles.Tabs,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: boxProps.className
  });
  var contextValue = React.useMemo(function () {
    return {
      tabs: tabs,
      overrides: overrides
    };
  }, [overrides, tabs]);
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, boxProps), {}, {
    className: className,
    children: /*#__PURE__*/React.createElement(TabsContext.Provider, {
      value: contextValue
    }, children)
  });
}, {
  themeKey: 'Tabs'
});
var Tabs = utils_createComponent.createComponent(function (props) {
  var tabsProps = useProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: props.use,
    htmlProps: tabsProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'Tabs'
  },
  themeKey: 'Tabs'
});

exports.Tabs = Tabs;
exports.TabsContext = TabsContext;
