// bumbag/es/Popover/Popover.styles.js
import 'react';
import '../Provider/ThemeContext.js';
import 'classnames';
import 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import '../utils/useTheme.js';
import '../utils/useLocalStorage.js';
import '../_rollupPluginBabelHelpers.js';
import '../utils/omit.js';
import '../ColorMode/utils.js';
import '../ColorMode/ColorModeContext.js';
import '../utils/pick.js';
import '../utils/cssProps.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import '../utils/isFunction.js';
import '../utils/get.js';
import '../getCSSFromStyleObject.js';
import 'lodash/uniq';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '../utils/getHiddenScrollbarStyles.js';
import '@emotion/is-prop-valid';
import '../utils/OutsideClickHandler.js';
import '../Modal.styles.js';
export { a as Popover, g as PopoverArrow, h as PopoverBackdrop, b as PopoverClose, c as PopoverContent, i as PopoverDisclosure, f as PopoverFooter, d as PopoverHeader, e as PopoverTitle, j as getPlacementAttributes } from '../Popover.styles.js';
