// bumbag/lib/theme/palette.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
require('reakit/Id');
var utils_colors = require('../utils/colors.js');
var utils_palette = require('../utils/palette.js');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('../utils/OutsideClickHandler.js');

var defaultPalette = {
  background: 'white',
  black: '#0b0e13',
  text: '#212121',
  primary: '#574feb',
  secondary: '#9e46d8',
  info: '#1e67d5',
  success: '#0a7d33',
  danger: '#da1717',
  warning: '#ed9c22',
  gray: '#626f84'
};
var defaultDarkPalette = {
  background: '#1a212d',
  black: '#0b0e13',
  default: '#27303f',
  defaultInverted: 'white',
  text: 'white',
  text100: '#b1bccd',
  text200: '#c1c9d7',
  text300: '#d0d7e1',
  text400: '#e0e4eb',
  primary: '#574feb',
  secondary: '#9e46d8',
  info: '#1e67d5',
  success: '#0a7d33',
  danger: '#da1717',
  warning: '#ed9c22'
};
var getDefaultPalette = (function (overrides) {
  var _overrides$text;

  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, utils_palette.generateTextVariants((_overrides$text = overrides === null || overrides === void 0 ? void 0 : overrides.text) !== null && _overrides$text !== void 0 ? _overrides$text : defaultPalette.text)), {}, {
    background: defaultPalette.background,
    black: defaultPalette.black,
    black500: defaultPalette.black,
    black400: utils_colors.tint(0.05, defaultPalette.black)({
      backgroundColor: overrides.background || defaultPalette.background,
      colorMode: 'default'
    }),
    black300: utils_colors.tint(0.1, defaultPalette.black)({
      backgroundColor: overrides.background || defaultPalette.background,
      colorMode: 'default'
    }),
    black200: utils_colors.tint(0.15, defaultPalette.black)({
      backgroundColor: overrides.background || defaultPalette.background,
      colorMode: 'default'
    }),
    black100: utils_colors.tint(0.2, defaultPalette.black)({
      backgroundColor: overrides.background || defaultPalette.background,
      colorMode: 'default'
    }),
    white: 'white',
    white500: 'white',
    white600: '#f7f7fa',
    white700: '#f2f2f5',
    white800: '#e6e6eb',
    white900: '#d9d9e1',
    default: '#fcfcfd',
    defaultInverted: '#212121'
  }, utils_palette.generateColorVariants({
    paletteKey: 'primary',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.primary || defaultPalette.primary
  })), utils_palette.generateColorVariants({
    paletteKey: 'secondary',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.secondary || defaultPalette.secondary
  })), utils_palette.generateColorVariants({
    paletteKey: 'info',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.info || defaultPalette.info
  })), utils_palette.generateColorVariants({
    paletteKey: 'success',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.success || defaultPalette.success
  })), utils_palette.generateColorVariants({
    paletteKey: 'danger',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.danger || defaultPalette.danger
  })), utils_palette.generateColorVariants({
    paletteKey: 'warning',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.warning || defaultPalette.warning,
    paletteOverrides: function paletteOverrides(_ref) {
      var color = _ref.color;
      return {
        warningTintInverted: utils_colors.shade(0.7, color)()
      };
    }
  })), utils_palette.generateColorVariants({
    paletteKey: 'gray',
    backgroundColor: overrides.background || defaultPalette.background,
    color: overrides.gray || defaultPalette.gray,
    paletteOverrides: function paletteOverrides(_ref2) {
      var color = _ref2.color;
      return {
        gray700: utils_colors.shade(0.2, color)(),
        gray800: utils_colors.shade(0.3, color)(),
        gray900: utils_colors.shade(0.4, color)()
      };
    }
  })), {}, {
    modes: {
      dark: _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, defaultDarkPalette), utils_palette.generateColorVariants({
        paletteKey: 'primary',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.primary || defaultDarkPalette.primary
      })), utils_palette.generateColorVariants({
        paletteKey: 'secondary',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.secondary || defaultDarkPalette.secondary
      })), utils_palette.generateColorVariants({
        paletteKey: 'info',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.info || defaultDarkPalette.info
      })), utils_palette.generateColorVariants({
        paletteKey: 'success',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.success || defaultDarkPalette.success
      })), utils_palette.generateColorVariants({
        paletteKey: 'danger',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.danger || defaultDarkPalette.danger
      })), utils_palette.generateColorVariants({
        paletteKey: 'warning',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.warning || defaultDarkPalette.warning,
        paletteOverrides: function paletteOverrides(_ref3) {
          var color = _ref3.color;
          return {
            warningTintInverted: utils_colors.shade(0.7, color)()
          };
        }
      })), utils_palette.generateColorVariants({
        paletteKey: 'gray',
        colorMode: 'dark',
        backgroundColor: overrides.background || defaultDarkPalette.background,
        color: overrides.gray || defaultPalette.gray
      })), {}, {
        black: defaultPalette.black,
        black500: defaultPalette.black,
        black400: utils_colors.tint(0.25, defaultPalette.black)({
          backgroundColor: overrides.background || defaultDarkPalette.background,
          colorMode: 'default'
        }),
        black300: utils_colors.tint(0.5, defaultPalette.black)({
          backgroundColor: overrides.background || defaultDarkPalette.background,
          colorMode: 'default'
        }),
        black200: utils_colors.tint(0.75, defaultPalette.black)({
          backgroundColor: overrides.background || defaultDarkPalette.background,
          colorMode: 'default'
        }),
        black100: utils_colors.tint(1, defaultPalette.black)({
          backgroundColor: overrides.background || defaultDarkPalette.background,
          colorMode: 'default'
        })
      })
    }
  }, overrides);
});

exports.default = getDefaultPalette;
