// bumbag/lib/Page/PageContent.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('../utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
require('../utils/OutsideClickHandler.js');
var reakit = require('reakit');
require('../Box.styles-fefb7baf.js');
require('../Box/Box.js');
require('../Container.styles-9dcb1344.js');
var Container_Container = require('../Container/Container.js');
var Page_Page_styles = require('../Page.styles-99603079.js');
var Page_PageContentWrapper = require('./PageContentWrapper.js');

var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var overrides = props.overrides,
      wrapperProps = props.wrapperProps,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["overrides", "wrapperProps"]);

  var containerProps = Container_Container.Container.useProps(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, restProps), {}, {
    wrapElement: function wrapElement(element) {
      return /*#__PURE__*/React.createElement(Page_PageContentWrapper.PageContentWrapper, _rollupPluginBabelHelpers._extends({
        overrides: overrides
      }, wrapperProps), element);
    }
  }));
  var className = utils_useClassName.useClassName({
    style: Page_Page_styles.PageContent,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: containerProps.className
  });
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, containerProps), {}, {
    className: className
  });
}, {
  defaultProps: {
    breakpoint: 'tablet',
    isLayout: true
  },
  themeKey: 'PageContent'
});
var PageContent = utils_createComponent.createComponent(function (props) {
  var pageContentProps = useProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: props.use,
    htmlProps: pageContentProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'PageContent'
  },
  themeKey: 'PageContent'
});

exports.PageContent = PageContent;
