// bumbag/lib/Show/Show.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
var classNames = require('classnames');
var emotion = require('emotion');
var core = require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
var utils_theme = require('../getCSSFromStyleObject-d83e232a.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('deepmerge');
require('lodash/set');
require('capsize');
require('@emotion/is-prop-valid');
require('../utils/OutsideClickHandler.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function _templateObject2() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n      display: none;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _rollupPluginBabelHelpers._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var aboveBreakpoints = {
  mobile: 'tablet',
  tablet: 'desktop',
  desktop: 'widescreen',
  widescreen: 'fullHD'
};
function Show(props) {
  var above = props.above,
      below = props.below;
  var breakpoint;

  if (above) {
    breakpoint = "min-" + aboveBreakpoints[above];
  } else if (below) {
    breakpoint = "max-" + below;
  }

  var showClassName = utils_useClassName.useClassName({
    style: showStyle,
    styleProps: {
      breakpoint: breakpoint
    }
  });
  var children = transformChildren(props.children, showClassName, 0);
  return children;
}

function transformChildren(children, className, index) {
  if (typeof children === 'string') {
    return /*#__PURE__*/React.createElement("span", {
      key: index,
      className: className
    }, children);
  } else if (React.isValidElement(children)) {
    if (children.type.toString() === 'Symbol(react.fragment)') {
      return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, children), {}, {
        props: _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, children.props || {}), {}, {
          // @ts-ignore
          children: transformChildren((children.props || {}).children, className, 0)
        })
      });
    }

    return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, children), {}, {
      // @ts-ignore
      props: _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, children.props || {}), {}, {
        className: classNames__default['default'](children.props.className, className)
      })
    });
  } else if (Array.isArray(children)) {
    return children.map(function (child, index) {
      return transformChildren(child, className, index);
    });
  }

  return null;
}

var showStyle = function showStyle(styleProps) {
  return emotion.css(_templateObject(), utils_theme.breakpoint(styleProps.breakpoint, core.css(_templateObject2()), {
    show: true
  })(styleProps));
};

exports.Show = Show;
exports.showStyle = showStyle;
