// bumbag/es/Hide/Hide.js
import { createElement, isValidElement } from 'react';
import '../Provider/ThemeContext.js';
import classNames from 'classnames';
import { css } from 'emotion';
import { css as css$1 } from '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import '../utils/useTheme.js';
import '../utils/useLocalStorage.js';
import { _ as _objectSpread2, a as _taggedTemplateLiteralLoose } from '../_rollupPluginBabelHelpers.js';
import '../utils/omit.js';
import '../ColorMode/utils.js';
import '../ColorMode/ColorModeContext.js';
import '../utils/pick.js';
import '../utils/cssProps.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import '../utils/isFunction.js';
import '../utils/get.js';
import { j as breakpoint } from '../getCSSFromStyleObject.js';
import 'lodash/uniq';
import { useClassName } from '../utils/useClassName.js';
import 'reakit/Id';
import 'deepmerge';
import 'lodash/set';
import 'capsize';
import '@emotion/is-prop-valid';
import '../utils/OutsideClickHandler.js';

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      display: none !important;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var aboveBreakpoints = {
  mobile: 'tablet',
  tablet: 'desktop',
  desktop: 'widescreen',
  widescreen: 'fullHD'
};
function Hide(props) {
  var above = props.above,
      below = props.below;
  var breakpoint;

  if (above) {
    breakpoint = "min-" + aboveBreakpoints[above];
  } else if (below) {
    breakpoint = "max-" + below;
  }

  var hideClassName = useClassName({
    style: hideStyle,
    styleProps: {
      breakpoint: breakpoint
    }
  });
  var children = transformChildren(props.children, hideClassName, 0);
  return children;
}

function transformChildren(children, className, index) {
  if (typeof children === 'string') {
    return /*#__PURE__*/createElement("span", {
      key: index,
      className: className
    }, children);
  } else if (isValidElement(children)) {
    if (children.type.toString() === 'Symbol(react.fragment)') {
      return _objectSpread2(_objectSpread2({}, children), {}, {
        props: _objectSpread2(_objectSpread2({}, children.props || {}), {}, {
          // @ts-ignore
          children: transformChildren((children.props || {}).children, className, 0)
        })
      });
    }

    return _objectSpread2(_objectSpread2({}, children), {}, {
      // @ts-ignore
      props: _objectSpread2(_objectSpread2({}, children.props || {}), {}, {
        className: classNames(children.props.className, className)
      })
    });
  } else if (Array.isArray(children)) {
    return children.map(function (child, index) {
      return transformChildren(child, className, index);
    });
  }

  return null;
}

var hideStyle = function hideStyle(styleProps) {
  return css(_templateObject(), breakpoint(styleProps.breakpoint, css$1(_templateObject2()))(styleProps));
};

export { Hide, hideStyle };
