// bumbag/es/Modal/index.js
import 'react';
import '../Provider/ThemeContext.js';
import 'classnames';
import 'emotion';
import '@emotion/core';
import 'emotion-theming';
import '@emotion/styled';
import '../utils/useTheme.js';
import '../utils/useLocalStorage.js';
import '../_rollupPluginBabelHelpers.js';
import '../utils/omit.js';
import '../ColorMode/utils.js';
import '../ColorMode/ColorModeContext.js';
import '../utils/pick.js';
import '../utils/cssProps.js';
import 'lodash/kebabCase';
import 'tinycolor2';
import '../utils/isFunction.js';
import '../utils/get.js';
import '../getCSSFromStyleObject.js';
import '../utils/useStyle.js';
import '../utils/omitBy.js';
import '../utils/useDefaultProps.js';
import 'lodash/uniq';
import '../utils/useClassName.js';
import 'reakit/Id';
import '../utils/mergeRefs.js';
import 'deepmerge';
import 'lodash/set';
import '../utils/createComponent.js';
import '../utils/createElement.js';
import '../utils/createHook.js';
import 'capsize';
import '../utils/getHiddenScrollbarStyles.js';
import '@emotion/is-prop-valid';
import '../utils/htmlProps.js';
import '../utils/OutsideClickHandler.js';
import 'reakit';
import '../Box.styles.js';
import '../Box/Box.js';
export { M as modalStyles } from '../Modal.styles.js';
import { useModalState, ModalState } from './ModalState.js';
export { ModalContext, ModalState, useModalState } from './ModalState.js';
import { ModalBackdrop } from './ModalBackdrop.js';
export { ModalBackdrop } from './ModalBackdrop.js';
import { Modal as Modal$1 } from './Modal.js';
import { ModalDisclosure } from './ModalDisclosure.js';
export { ModalDisclosure } from './ModalDisclosure.js';

var Modal = Object.assign(Modal$1, {
  Backdrop: ModalBackdrop,
  Disclosure: ModalDisclosure,
  useState: useModalState,
  State: ModalState
});

export { Modal };
