// bumbag/lib/FieldWrapper/FieldWrapper.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
var utils_uniqueId = require('../utils/uniqueId.js');
require('../utils/parseIcons.js');
require('../utils/colors.js');
require('../utils/palette.js');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('../utils/getCapsizeStyles.js');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
var utils_OutsideClickHandler = require('../utils/OutsideClickHandler.js');
require('../theme/palette.js');
var reakit = require('reakit');
require('../Button.styles-ece38a73.js');
require('conditional-wrap');
require('../Box.styles-fefb7baf.js');
var Box_Box = require('../Box/Box.js');
require('../Flex.styles-db693d59.js');
require('../Flex/Flex.js');
require('../Icon.styles-1304be6b.js');
var Icon_Icon = require('../Icon/Icon.js');
require('../Text.styles-7cd48b0c.js');
require('../Text/Text.js');
require('../Text/TextBlock.js');
require('../Text/TextInline.js');
var Text_index = require('../Text/index.js');
require('../Spinner.styles-6992127d.js');
require('../Spinner/Spinner.js');
require('../Button/Button.js');
var VisuallyHidden = require('reakit/VisuallyHidden');
require('../Button/ButtonClose.js');
var Button_index = require('../Button/index.js');
var FieldWrapper_FieldWrapper_styles = require('../FieldWrapper.styles-c26b9d24.js');
require('../Card.styles-c2e6de71.js');
require('../Card/Card.js');
var Card_index = require('../Card/index.js');
require('../Disclosure.styles-69be3e1a.js');
require('../Disclosure/DisclosureState.js');
require('../Disclosure/Disclosure.js');
require('../Disclosure/DisclosureContent.js');
var Disclosure_index = require('../Disclosure/index.js');
require('../Label.styles-aef4bc27.js');
var Label_Label = require('../Label/Label.js');

var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var children = props.children,
      description = props.description,
      hint = props.hint,
      isOptional = props.isOptional,
      label = props.label,
      labelType = props.labelType,
      isRequired = props.isRequired,
      state = props.state,
      tooltip = props.tooltip,
      tooltipTriggerComponent = props.tooltipTriggerComponent,
      validationText = props.validationText,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["children", "description", "hint", "isOptional", "label", "labelType", "isRequired", "state", "tooltip", "tooltipTriggerComponent", "validationText"]);

  var boxProps = Box_Box.Box.useProps(restProps);
  var className = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.FieldWrapper,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: boxProps.className
  });
  var labelClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.Label,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'Label'
  });
  var descriptionClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.DescriptionText,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'DescriptionText'
  });
  var hintClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.HintText,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'HintText'
  });
  var optionalClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.OptionalText,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'OptionalText'
  });
  var requiredClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.RequiredText,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'RequiredText'
  });
  var validationClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.ValidationText,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'ValidationText'
  });
  var tooltipTriggerClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.TooltipTrigger,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'TooltipTrigger'
  });
  var tooltipPopoverClassName = utils_useClassName.useClassName({
    style: FieldWrapper_FieldWrapper_styles.TooltipPopover,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'TooltipPopover'
  });
  var disclosure = Disclosure_index.Disclosure.useState();
  var uid = utils_uniqueId.useUniqueId();
  var elementProps = {
    isRequired: isRequired,
    id: uid,
    state: state
  };
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, boxProps), {}, {
    className: className,
    children: /*#__PURE__*/React.createElement(React.Fragment, null, label && /*#__PURE__*/React.createElement(Box_Box.Box, {
      marginBottom: "minor-3"
    }, /*#__PURE__*/React.createElement(Box_Box.Box, {
      display: "flex",
      alignItems: "center",
      lineHeight: "none"
    }, typeof label === 'string' ? /*#__PURE__*/React.createElement(React.Fragment, null, labelType === 'legend' ? /*#__PURE__*/React.createElement(Label_Label.Label, {
      use: "legend",
      className: labelClassName
    }, label) :
    /*#__PURE__*/
    // @ts-ignore
    React.createElement(Label_Label.Label, {
      className: labelClassName,
      htmlFor: uid
    }, label)) : label, isOptional && /*#__PURE__*/React.createElement(Box_Box.Box, {
      className: optionalClassName
    }, "OPTIONAL"), isRequired && /*#__PURE__*/React.createElement(Box_Box.Box, {
      className: requiredClassName
    }, "*"), tooltip && /*#__PURE__*/React.createElement(Box_Box.Box, {
      position: "relative",
      marginLeft: "minor-2"
    }, /*#__PURE__*/React.createElement(utils_OutsideClickHandler.OutsideClickHandler, {
      onOutsideClick: disclosure.hide
    }, tooltipTriggerComponent ? React.cloneElement(tooltipTriggerComponent, {
      onClick: disclosure.toggle
    }) : /*#__PURE__*/React.createElement(Button_index.Button, {
      className: tooltipTriggerClassName,
      onClick: disclosure.toggle
    }, /*#__PURE__*/React.createElement(VisuallyHidden.VisuallyHidden, null, "Toggle tooltip"), /*#__PURE__*/React.createElement(Icon_Icon.Icon, {
      "aria-hidden": true,
      icon: "question-circle",
      verticalAlign: "-0.125rem"
    }))), disclosure.visible && /*#__PURE__*/React.createElement(Card_index.Card, {
      className: tooltipPopoverClassName
    }, typeof tooltip === 'string' ? /*#__PURE__*/React.createElement(Text_index.Text, {
      fontSize: "150"
    }, tooltip) : tooltip))), description && /*#__PURE__*/React.createElement(Box_Box.Box, {
      marginTop: "minor-1"
    }, typeof description === 'string' ? /*#__PURE__*/React.createElement(Text_index.Text.Block, {
      className: descriptionClassName
    }, description) : description)), typeof children === 'function' ?
    /*
    // @ts-ignore */
    children({
      elementProps: elementProps
    }) : React.cloneElement(children, elementProps), hint && /*#__PURE__*/React.createElement(Box_Box.Box, {
      marginTop: "minor-1"
    }, typeof hint === 'string' ? /*#__PURE__*/React.createElement(Box_Box.Box, {
      className: hintClassName
    }, hint) : hint), validationText && /*#__PURE__*/React.createElement(Box_Box.Box, {
      className: validationClassName,
      color: state
    }, validationText))
  });
}, {
  themeKey: 'FieldWrapper'
});
var FieldWrapper = utils_createComponent.createComponent(function (props) {
  var textProps = useProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    enableRenderPropsComposition: false,
    use: props.use,
    htmlProps: textProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'FieldWrapper'
  },
  themeKey: 'FieldWrapper'
});

exports.FieldWrapper = FieldWrapper;
