// bumbag/lib/ColorMode/ColorModeContext.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var utils_useTheme = require('../utils/useTheme.js');
var utils_useLocalStorage = require('../utils/useLocalStorage.js');
require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
var ColorMode_utils = require('./utils.js');

var ColorModeContext = React.createContext({
  setColorMode: function setColorMode() {},
  colorMode: 'default'
});
ColorModeProvider.defaultProps = {
  isSSR: false
};
function ColorModeProvider(props) {
  var children = props.children,
      _defaultMode = props.mode,
      isSSR = props.isSSR; ////////////////////////////////////

  var _useTheme = utils_useTheme.useTheme(),
      theme = _useTheme.theme;

  var localStorage = utils_useLocalStorage.useLocalStorage();
  var defaultMode = React.useMemo(function () {
    return ColorMode_utils.getDefaultColorMode(_defaultMode, {
      localStorage: localStorage,
      theme: theme
    });
  }, [_defaultMode]); // eslint-disable-line
  ////////////////////////////////////

  var _React$useState = React.useState(defaultMode),
      mode = _React$useState[0],
      setMode = _React$useState[1]; ////////////////////////////////////


  React.useEffect(function () {
    ColorMode_utils.addColorModeBodyClassName(defaultMode);
    localStorage.set('mode', defaultMode);
  }, [defaultMode]); // eslint-disable-line
  ////////////////////////////////////

  var setColorMode = React.useCallback(function (colorMode) {
    ColorMode_utils.addColorModeBodyClassName(colorMode, mode);
    localStorage.set('mode', colorMode);
    setMode(colorMode);
  }, [localStorage, mode]); ////////////////////////////////////

  var value = React.useMemo(function () {
    return {
      colorMode: mode,
      setColorMode: setColorMode
    };
  }, [mode, setColorMode]); ////////////////////////////////////

  var _React$useState2 = React.useState(false),
      mounted = _React$useState2[0],
      setMounted = _React$useState2[1];

  var useIsomorphicLayoutEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
  useIsomorphicLayoutEffect(function () {
    if (typeof window !== 'undefined') {
      setMounted(true);
    }
  }, []); ////////////////////////////////////

  if (isSSR && !mounted) {
    return /*#__PURE__*/React.createElement("div", {
      style: {
        visibility: 'hidden'
      }
    }, children);
  }

  return /*#__PURE__*/React.createElement(ColorModeContext.Provider, {
    value: value
  }, children);
}
function useColorMode() {
  return React.useContext(ColorModeContext);
}

exports.ColorModeContext = ColorModeContext;
exports.ColorModeProvider = ColorModeProvider;
exports.useColorMode = useColorMode;
