// bumbag/lib/Input/Input.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
var utils_cssProps = require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('../utils/uniqueId.js');
require('../utils/parseIcons.js');
require('../utils/colors.js');
require('../utils/palette.js');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('../utils/getCapsizeStyles.js');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
require('../utils/OutsideClickHandler.js');
require('../theme/palette.js');
var reakit = require('reakit');
require('../Button.styles-ece38a73.js');
var ConditionalWrap = require('conditional-wrap');
require('../Box.styles-fefb7baf.js');
var Box_Box = require('../Box/Box.js');
require('../Flex.styles-db693d59.js');
require('../Flex/Flex.js');
require('../Icon.styles-1304be6b.js');
var Icon_Icon = require('../Icon/Icon.js');
require('../Text.styles-7cd48b0c.js');
require('../Text/Text.js');
require('../Text/TextBlock.js');
require('../Text/TextInline.js');
require('../Text/index.js');
require('../Spinner.styles-6992127d.js');
var Spinner_Spinner = require('../Spinner/Spinner.js');
require('../Button/Button.js');
require('reakit/VisuallyHidden');
require('../Button/ButtonClose.js');
require('../Button/index.js');
var Input_Input_styles = require('../Input.styles-8ea80a0d.js');
var InputMask = require('react-input-mask');
require('../FieldWrapper.styles-c26b9d24.js');
require('../Card.styles-c2e6de71.js');
require('../Card/Card.js');
require('../Card/index.js');
require('../Disclosure.styles-69be3e1a.js');
require('../Disclosure/DisclosureState.js');
require('../Disclosure/Disclosure.js');
require('../Disclosure/DisclosureContent.js');
require('../Disclosure/index.js');
require('../Label.styles-aef4bc27.js');
require('../Label/Label.js');
var FieldWrapper_FieldWrapper = require('../FieldWrapper/FieldWrapper.js');
require('../Group.styles-79833991.js');
var Group_Group = require('../Group/Group.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var ConditionalWrap__default = /*#__PURE__*/_interopDefaultLegacy(ConditionalWrap);
var InputMask__default = /*#__PURE__*/_interopDefaultLegacy(InputMask);

var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var before = props.before,
      after = props.after,
      inputProps = props.inputProps,
      inputRef = props.inputRef,
      isLoading = props.isLoading,
      isRequired = props.isRequired,
      state = props.state,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["before", "after", "inputProps", "inputRef", "isLoading", "isRequired", "state"]);

  var wrapperClassName = utils_useClassName.useClassName({
    style: Input_Input_styles.InputWrapper,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'Wrapper',
    prevClassName: restProps.className
  });
  var spinnerClassName = utils_useClassName.useClassName({
    style: Input_Input_styles.InputSpinner,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'Spinner'
  });
  var boxProps = Box_Box.Box.useProps(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, utils_cssProps.omitCSSProps(restProps)), inputProps), {}, {
    className: undefined,
    elementRef: inputRef || props.elementRef,
    wrapElement: function wrapElement(children) {
      return /*#__PURE__*/React.createElement(Box_Box.Box, _rollupPluginBabelHelpers._extends({
        className: wrapperClassName
      }, utils_cssProps.pickCSSProps(props)), before && /*#__PURE__*/React.createElement(Box_Box.Box, {
        display: "inline-flex",
        position: "absolute",
        zIndex: "3",
        height: "100%"
      }, before), after && /*#__PURE__*/React.createElement(Box_Box.Box, {
        display: "inline-flex",
        position: "absolute",
        right: "0",
        zIndex: "3",
        height: "100%"
      }, after), children, isLoading && /*#__PURE__*/React.createElement(Spinner_Spinner.Spinner, {
        className: spinnerClassName,
        color: "text"
      }));
    }
  }));
  var className = utils_useClassName.useClassName({
    style: Input_Input_styles.Input,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: boxProps.className
  });
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, boxProps), {}, {
    className: className,
    'aria-invalid': state === 'danger',
    'aria-required': isRequired
  });
}, {
  defaultProps: {
    type: 'text'
  },
  themeKey: 'Input'
});
var Input = utils_createComponent.createComponent(function (props) {
  var inputProps = useProps(props);
  var use = props.use;

  if (props.mask) {
    use = InputMask__default['default'];
  }

  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: use,
    htmlProps: _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, inputProps), props.mask ? {
      mask: props.mask
    } : {})
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'Input'
  },
  defaultProps: {
    use: 'input'
  },
  themeKey: 'Input'
}); ////////////////////////////////////////////////////////////////

function InputIcon(props) {
  var className = utils_useClassName.useClassName({
    style: Input_Input_styles.InputIcon,
    styleProps: props,
    themeKey: 'Input.Icon',
    prevClassName: props.className
  });
  return /*#__PURE__*/React.createElement(Icon_Icon.Icon, _rollupPluginBabelHelpers._extends({
    color: "gray200"
  }, props, {
    className: className
  }));
} ////////////////////////////////////////////////////////////////

var useInputFieldProps = utils_createHook.createHook(function (props, _ref2) {
  var themeKey = _ref2.themeKey;

  var addonAfter = props.addonAfter,
      addonBefore = props.addonBefore,
      children = props.children,
      after = props.after,
      autoComplete = props.autoComplete,
      autoFocus = props.autoFocus,
      before = props.before,
      defaultValue = props.defaultValue,
      description = props.description,
      disabled = props.disabled,
      hint = props.hint,
      inputRef = props.inputRef,
      inputProps = props.inputProps,
      isLoading = props.isLoading,
      isOptional = props.isOptional,
      isRequired = props.isRequired,
      orientation = props.orientation,
      label = props.label,
      name = props.name,
      size = props.size,
      mask = props.mask,
      max = props.max,
      maxLength = props.maxLength,
      min = props.min,
      minLength = props.minLength,
      multiple = props.multiple,
      pattern = props.pattern,
      placeholder = props.placeholder,
      readOnly = props.readOnly,
      spellCheck = props.spellCheck,
      step = props.step,
      state = props.state,
      tooltip = props.tooltip,
      tooltipTriggerComponent = props.tooltipTriggerComponent,
      type = props.type,
      value = props.value,
      onBlur = props.onBlur,
      onChange = props.onChange,
      onFocus = props.onFocus,
      overrides = props.overrides,
      validationText = props.validationText,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["addonAfter", "addonBefore", "children", "after", "autoComplete", "autoFocus", "before", "defaultValue", "description", "disabled", "hint", "inputRef", "inputProps", "isLoading", "isOptional", "isRequired", "orientation", "label", "name", "size", "mask", "max", "maxLength", "min", "minLength", "multiple", "pattern", "placeholder", "readOnly", "spellCheck", "step", "state", "tooltip", "tooltipTriggerComponent", "type", "value", "onBlur", "onChange", "onFocus", "overrides", "validationText"]);

  var boxProps = Box_Box.Box.useProps(restProps);
  var className = utils_useClassName.useClassName({
    style: Input_Input_styles.InputField,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: boxProps.className
  });
  var groupClassName = utils_useClassName.useClassName({
    style: Input_Input_styles.InputFieldGroup,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'Group'
  });
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, boxProps), {}, {
    className: className,
    children: /*#__PURE__*/React.createElement(FieldWrapper_FieldWrapper.FieldWrapper, {
      description: description,
      hint: hint,
      isOptional: isOptional,
      isRequired: isRequired,
      label: label,
      overrides: overrides,
      state: state,
      tooltip: tooltip,
      tooltipTriggerComponent: tooltipTriggerComponent,
      validationText: validationText
    }, function (_ref3) {
      var elementProps = _ref3.elementProps;
      return /*#__PURE__*/React.createElement(ConditionalWrap__default['default'], {
        condition: Boolean(addonBefore || addonAfter),
        wrap: function wrap(children) {
          return /*#__PURE__*/React.createElement(Group_Group.Group, {
            className: groupClassName,
            orientation: orientation,
            overrides: overrides
          }, children);
        }
      }, /*#__PURE__*/React.createElement(React.Fragment, null, addonBefore, /*#__PURE__*/React.createElement(Input, _rollupPluginBabelHelpers._extends({
        after: after,
        autoComplete: autoComplete,
        autoFocus: autoFocus,
        before: before,
        defaultValue: defaultValue,
        disabled: disabled,
        inputProps: inputProps,
        inputRef: inputRef,
        isLoading: isLoading,
        isRequired: isRequired,
        name: name,
        size: size,
        mask: mask,
        max: max,
        maxLength: maxLength,
        min: min,
        minLength: minLength,
        multiple: multiple,
        pattern: pattern,
        placeholder: placeholder,
        readOnly: readOnly,
        spellCheck: spellCheck,
        step: step,
        state: state,
        type: type,
        value: value,
        onBlur: onBlur,
        onChange: onChange,
        onFocus: onFocus,
        overrides: overrides
      }, elementProps)), addonAfter));
    })
  });
}, {
  defaultProps: {
    orientation: 'horizontal'
  },
  themeKey: 'InputField'
});
var InputField = utils_createComponent.createComponent(function (props) {
  var inputFieldProps = useInputFieldProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: props.use,
    htmlProps: inputFieldProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'InputField'
  },
  themeKey: 'InputField'
});

exports.Input = Input;
exports.InputField = InputField;
exports.InputIcon = InputIcon;
