// bumbag/lib/DropdownMenu/DropdownMenuOptionGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
var utils_useOptionsState = require('../utils/useOptionsState.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('../utils/uniqueId.js');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('../utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
require('../utils/OutsideClickHandler.js');
var reakit = require('reakit');
require('../Box.styles-fefb7baf.js');
require('../Box/Box.js');
require('../Menu.styles-ef47d2ca.js');
var DropdownMenu_DropdownMenu_styles = require('../DropdownMenu.styles-2108c433.js');
require('./DropdownMenuButton.js');
require('../Modal.styles-6ced861b.js');
require('../Popover.styles-a83283b6.js');
require('./DropdownMenuPopover.js');
require('./DropdownMenuState.js');
require('./DropdownMenu.js');
var DropdownMenu_DropdownMenuGroup = require('./DropdownMenuGroup.js');

var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var children = props.children,
      defaultValue = props.defaultValue,
      onBlur = props.onBlur,
      onChange = props.onChange,
      type = props.type,
      value = props.value,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["children", "defaultValue", "onBlur", "onChange", "type", "value"]);

  var _useOptionsState = utils_useOptionsState.useOptionsState({
    defaultValue: defaultValue,
    onBlur: onBlur,
    onChange: onChange,
    type: type,
    value: value
  }),
      getOptionItemProps = _useOptionsState.getOptionItemProps;

  var dropdownMenuGroupProps = DropdownMenu_DropdownMenuGroup.DropdownMenuGroup.useProps(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, restProps), {}, {
    children: React.Children.map(children, function (child) {
      var _child$props;

      if (!React.isValidElement(child)) return child;
      return React.cloneElement(child, _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, child.props), getOptionItemProps({
        value: (_child$props = child.props) === null || _child$props === void 0 ? void 0 : _child$props.value
      })));
    })
  }));
  var className = utils_useClassName.useClassName({
    style: DropdownMenu_DropdownMenu_styles.DropdownMenuOptionGroup,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: dropdownMenuGroupProps.className
  });
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, dropdownMenuGroupProps), {}, {
    className: className
  });
}, {
  themeKey: 'DropdownMenu.OptionGroup'
});
var DropdownMenuOptionGroup = utils_createComponent.createComponent(function (props) {
  var htmlProps = useProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: props.use,
    htmlProps: htmlProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'DropdownMenu.OptionGroup'
  },
  themeKey: 'DropdownMenu.OptionGroup'
});

exports.DropdownMenuOptionGroup = DropdownMenuOptionGroup;
