// bumbag/lib/DropdownMenu/DropdownMenuOptionItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
require('../utils/useTheme.js');
require('../utils/useLocalStorage.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('../utils/omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('../utils/pick.js');
require('../utils/cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('../utils/isFunction.js');
require('../utils/get.js');
require('../getCSSFromStyleObject-d83e232a.js');
require('../utils/useStyle.js');
require('../utils/omitBy.js');
require('../utils/useDefaultProps.js');
require('lodash/uniq');
var utils_useClassName = require('../utils/useClassName.js');
require('reakit/Id');
require('../utils/parseIcons.js');
require('../utils/mergeRefs.js');
require('deepmerge');
require('lodash/set');
require('../utils/bindFns.js');
var utils_createComponent = require('../utils/createComponent.js');
var utils_createElement = require('../utils/createElement.js');
var utils_createHook = require('../utils/createHook.js');
require('capsize');
require('../utils/getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
require('../utils/htmlProps.js');
require('../utils/OutsideClickHandler.js');
var reakit = require('reakit');
require('../Box.styles-fefb7baf.js');
var Box_Box = require('../Box/Box.js');
require('../Icon.styles-1304be6b.js');
var Icon_Icon = require('../Icon/Icon.js');
require('../Menu.styles-ef47d2ca.js');
var DropdownMenu_DropdownMenu_styles = require('../DropdownMenu.styles-2108c433.js');
require('./DropdownMenuButton.js');
require('../Modal.styles-6ced861b.js');
require('../Popover.styles-a83283b6.js');
require('./DropdownMenuPopover.js');
require('./DropdownMenuState.js');
require('./DropdownMenu.js');
var DropdownMenu_DropdownMenuItem = require('./DropdownMenuItem.js');

var useProps = utils_createHook.createHook(function (props, _ref) {
  var themeKey = _ref.themeKey;

  var children = props.children,
      checked = props.checked,
      onChange = props.onChange,
      value = props.value,
      restProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(props, ["children", "checked", "onChange", "value"]); //////////////////////////////////////////////////////


  var menuItemProps = DropdownMenu_DropdownMenuItem.DropdownMenuItem.useProps(restProps); //////////////////////////////////////////////////////

  var className = utils_useClassName.useClassName({
    style: DropdownMenu_DropdownMenu_styles.DropdownMenuOptionItem,
    styleProps: props,
    themeKey: themeKey,
    prevClassName: menuItemProps.className
  });
  var iconWrapperClassName = utils_useClassName.useClassName({
    style: DropdownMenu_DropdownMenu_styles.DropdownMenuOptionItemIconWrapper,
    styleProps: props,
    themeKey: themeKey,
    themeKeySuffix: 'IconWrapper'
  }); //////////////////////////////////////////////////////

  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, menuItemProps), {}, {
    'aria-checked': checked,
    className: className,
    onClick: function onClick() {
      return onChange && onChange({
        checked: !checked,
        value: value
      });
    },
    children: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Box_Box.Box, {
      className: iconWrapperClassName,
      overrides: props.overrides
    }, checked && /*#__PURE__*/React.createElement(Icon_Icon.Icon, {
      icon: {
        viewBoxWidth: 16,
        viewBoxHeight: 16,
        paths: ['M14 3c-.28 0-.53.11-.71.29L6 10.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l8-8A1.003 1.003 0 0014 3z']
      },
      overrides: props.overrides
    })), children)
  });
}, {
  themeKey: 'DropdownMenu.OptionItem'
});
var DropdownMenuOptionItem = utils_createComponent.createComponent(function (props) {
  var htmlProps = useProps(props);
  return utils_createElement.createElement({
    children: props.children,
    component: reakit.Box,
    use: props.use,
    htmlProps: htmlProps
  });
}, {
  attach: {
    useProps: useProps,
    displayName: 'DropdownMenu.OptionItem'
  },
  themeKey: 'DropdownMenu.OptionItem'
});

exports.DropdownMenuOptionItem = DropdownMenuOptionItem;
