// bumbag/lib/utils/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
var utils_debounceAnimationFrame = require('./debounceAnimationFrame.js');
var utils_useViewportHeight = require('./useViewportHeight.js');
require('../Provider/ThemeContext.js');
require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var utils_useTheme = require('./useTheme.js');
require('./useLocalStorage.js');
require('../_rollupPluginBabelHelpers-41d6b37b.js');
var utils_omit = require('./omit.js');
require('../ColorMode/utils.js');
require('../ColorMode/ColorModeContext.js');
require('./pick.js');
var utils_cssProps = require('./cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
var utils_isFunction = require('./isFunction.js');
require('./get.js');
var utils_theme = require('../getCSSFromStyleObject-d83e232a.js');
var utils_useStyle = require('./useStyle.js');
var utils_usePrevious = require('./usePrevious.js');
var utils_useOptionsState = require('./useOptionsState.js');
require('./omitBy.js');
require('./useDefaultProps.js');
var utils_useDebounce = require('./useDebounce.js');
require('lodash/uniq');
var utils_useClassName = require('./useClassName.js');
var utils_getMediaQueryList = require('./getMediaQueryList.js');
var utils_useBreakpointValue = require('./useBreakpointValue.js');
var utils_useBreakpoint = require('./useBreakpoint.js');
require('reakit/Id');
var utils_uniqueId = require('./uniqueId.js');
var utils_times = require('./times.js');
var utils_parseIcons = require('./parseIcons.js');
var utils_colors = require('./colors.js');
var utils_palette = require('./palette.js');
var utils_mergeRefs = require('./mergeRefs.js');
var deepmerge = require('deepmerge');
require('lodash/set');
var utils_applyTheme = require('./applyTheme.js');
var utils_bindFns = require('./bindFns.js');
var utils_createComponent = require('./createComponent.js');
var utils_createElement = require('./createElement.js');
var utils_createHook = require('./createHook.js');
var utils_forwardRefWithUse = require('./forwardRefWithUse.js');
var utils_fieldAdaptors = require('./fieldAdaptors.js');
require('capsize');
var utils_getCapsizeStyles = require('./getCapsizeStyles.js');
var utils_getHiddenScrollbarStyles = require('./getHiddenScrollbarStyles.js');
require('@emotion/is-prop-valid');
var utils_htmlProps = require('./htmlProps.js');
var utils_OutsideClickHandler = require('./OutsideClickHandler.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var deepmerge__default = /*#__PURE__*/_interopDefaultLegacy(deepmerge);



exports.debounceAnimationFrame = utils_debounceAnimationFrame.debounceAnimationFrame;
exports.useViewportHeight = utils_useViewportHeight.useViewportHeight;
exports.useTheme = utils_useTheme.useTheme;
exports.omit = utils_omit.omit;
exports.cssProps = utils_cssProps.cssProps;
exports.isCSSProp = utils_cssProps.isCSSProp;
exports.omitCSSProps = utils_cssProps.omitCSSProps;
exports.pickCSSProps = utils_cssProps.pickCSSProps;
exports.isFunction = utils_isFunction.isFunction;
exports.altitude = utils_theme.altitude;
exports.border = utils_theme.border;
exports.borderRadius = utils_theme.borderRadius;
exports.breakpoint = utils_theme.breakpoint;
exports.font = utils_theme.font;
exports.fontMetric = utils_theme.fontMetric;
exports.fontSize = utils_theme.fontSize;
exports.fontWeight = utils_theme.fontWeight;
exports.letterSpacing = utils_theme.letterSpacing;
exports.lineHeight = utils_theme.lineHeight;
exports.palette = utils_theme.palette;
exports.space = utils_theme.space;
exports.theme = utils_theme.theme;
exports.useStyle = utils_useStyle.useStyle;
exports.usePrevious = utils_usePrevious.usePrevious;
exports.useOptionsState = utils_useOptionsState.useOptionsState;
exports.useDebounce = utils_useDebounce.useDebounce;
exports.useClassName = utils_useClassName.useClassName;
exports.getMediaQueryList = utils_getMediaQueryList.getMediaQueryList;
exports.useBreakpointValue = utils_useBreakpointValue.useBreakpointValue;
exports.useBreakpoint = utils_useBreakpoint.useBreakpoint;
exports.generateId = utils_uniqueId.generateId;
exports.useUniqueId = utils_uniqueId.useUniqueId;
exports.times = utils_times.times;
exports.parseIcons = utils_parseIcons.parseIcons;
exports.darken = utils_colors.darken;
exports.isColor = utils_colors.isColor;
exports.lighten = utils_colors.lighten;
exports.readableColor = utils_colors.readableColor;
exports.shade = utils_colors.shade;
exports.tint = utils_colors.tint;
exports.generateColorVariants = utils_palette.generateColorVariants;
exports.generateTextVariants = utils_palette.generateTextVariants;
exports.mergeRefs = utils_mergeRefs.mergeRefs;
Object.defineProperty(exports, 'merge', {
	enumerable: true,
	get: function () {
		return deepmerge__default['default'];
	}
});
exports.applyTheme = utils_applyTheme.applyTheme;
exports.bindFns = utils_bindFns.bindFns;
exports.createComponent = utils_createComponent.createComponent;
exports.createElement = utils_createElement.createElement;
exports.createHook = utils_createHook.createHook;
exports.forwardRefWithUse = utils_forwardRefWithUse.forwardRefWithUse;
exports.formikField = utils_fieldAdaptors.formikField;
exports.reduxFormField = utils_fieldAdaptors.reduxFormField;
exports.getCapsizeAttributes = utils_getCapsizeStyles.getCapsizeAttributes;
exports.getCapsizeStyles = utils_getCapsizeStyles.getCapsizeStyles;
exports.getFontSize = utils_getCapsizeStyles.getFontSize;
exports.getHiddenScrollbarStyles = utils_getHiddenScrollbarStyles.getHiddenScrollbarStyles;
exports.pickHTMLProps = utils_htmlProps.pickHTMLProps;
exports.OutsideClickHandler = utils_OutsideClickHandler.OutsideClickHandler;
