// bumbag/lib/utils/palette.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('classnames');
require('emotion');
require('@emotion/core');
require('emotion-theming');
require('@emotion/styled');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-41d6b37b.js');
require('./omit.js');
require('../ColorMode/utils.js');
require('./pick.js');
require('./cssProps.js');
require('lodash/kebabCase');
require('tinycolor2');
require('./isFunction.js');
require('./get.js');
require('../getCSSFromStyleObject-d83e232a.js');
var utils_colors = require('./colors.js');

var generateTextVariants = function generateTextVariants(textColor) {
  return {
    text100: utils_colors.lighten(0.2, textColor)(),
    text200: utils_colors.lighten(0.15, textColor)(),
    text300: utils_colors.lighten(0.1, textColor)(),
    text400: utils_colors.lighten(0.05, textColor)(),
    text: textColor,
    textTint: utils_colors.tint(0.8, textColor)(),
    textInverted: utils_colors.readableColor(textColor)(),
    textTintInverted: utils_colors.shade(0.3, textColor)()
  };
};
var generateColorVariants = function generateColorVariants(_ref) {
  var _objectSpread2;

  var backgroundColor = _ref.backgroundColor,
      color = _ref.color,
      _ref$colorMode = _ref.colorMode,
      colorMode = _ref$colorMode === void 0 ? 'default' : _ref$colorMode,
      paletteKey = _ref.paletteKey,
      paletteOverrides = _ref.paletteOverrides;
  return _rollupPluginBabelHelpers._objectSpread2((_objectSpread2 = {}, _objectSpread2[paletteKey + "100"] = utils_colors.tint(0.7, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "200"] = utils_colors.tint(0.5, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "300"] = utils_colors.tint(0.3, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "400"] = utils_colors.tint(0.1, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey] = color, _objectSpread2[paletteKey + "500"] = color, _objectSpread2[paletteKey + "600"] = utils_colors.shade(0.1, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "700"] = utils_colors.shade(0.3, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "800"] = utils_colors.shade(0.5, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "900"] = utils_colors.shade(0.7, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "Shade"] = utils_colors.shade(0.8, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "Tint"] = utils_colors.tint(0.9, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "Inverted"] = utils_colors.readableColor(color)(), _objectSpread2[paletteKey + "ShadeInverted"] = utils_colors.tint(0.7, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2[paletteKey + "TintInverted"] = utils_colors.shade(0.5, color)({
    backgroundColor: backgroundColor,
    colorMode: colorMode
  }), _objectSpread2), paletteOverrides ? paletteOverrides({
    color: color
  }) : {});
};

exports.generateColorVariants = generateColorVariants;
exports.generateTextVariants = generateTextVariants;
